// @jeecg/online@3.6.2-beta/auth.data-8be5c9e3.mjs
import { computed as s } from "vue";
const n = [
  {
    title: "启用",
    dataIndex: "switch",
    width: 80,
    align: "center",
    slots: { customRender: "switch" }
  },
  {
    title: "字段名称",
    width: 200,
    dataIndex: "code"
  },
  {
    title: "字段描述",
    // width: 200,
    dataIndex: "title"
  },
  {
    title: "列表控制",
    dataIndex: "list",
    width: 120,
    slots: { customRender: "list" }
  },
  {
    title: "表单控制",
    dataIndex: "form",
    width: 180,
    slots: { customRender: "form" }
  }
], i = [
  {
    title: "启用",
    dataIndex: "switch",
    width: 80,
    slots: { customRender: "switch" }
  },
  {
    title: "名称",
    dataIndex: "title"
  },
  {
    title: "编码",
    dataIndex: "code"
  },
  {
    title: "权限控制",
    dataIndex: "control",
    width: 180,
    slots: { customRender: "control" }
  }
], u = [
  { code: "add", title: "新增", status: 0 },
  { code: "update", title: "编辑", status: 0 },
  { code: "delete", title: "删除", status: 0 },
  { code: "batch_delete", title: "批量删除", status: 0 },
  { code: "export", title: "导出", status: 0 },
  { code: "import", title: "导入", status: 0 },
  { code: "bpm", title: "提交流程", status: 0 },
  { code: "super_query", title: "高级查询", status: 0 }
], a = "USE_SQL_RULES", r = [
  {
    title: "启用",
    dataIndex: "switch",
    width: 80,
    slots: { customRender: "switch" }
  },
  {
    title: "规则名称",
    dataIndex: "ruleName",
    width: 130
  },
  {
    title: "规则描述",
    dataIndex: "description",
    customRender({ record: { ruleOperator: t, ruleValue: e, ruleColumn: d } }) {
      return t == a ? `自定义SQL: ${e}` : `${d} ${t} ${e}`;
    }
  }
];
function c(t, e) {
  return { formSchemas: s(() => [
    {
      label: "规则名称",
      field: "ruleName",
      required: !0,
      component: "Input"
    },
    {
      label: "规则字段",
      field: "ruleColumn",
      component: "JSearchSelect",
      componentProps: {
        dictOptions: t.authFields,
        getPopupContainer: () => document.body
      },
      dynamicRules({ model: o }) {
        return [{ required: o.ruleOperator != a, message: "请输入规则字段" }];
      },
      show: ({ model: o }) => o.ruleOperator != a
    },
    {
      label: "条件规则",
      field: "ruleOperator",
      required: !0,
      component: "JDictSelectTag",
      componentProps: {
        dictCode: "rule_conditions",
        onChange: e.onRuleOperatorChange
      }
    },
    {
      label: "规则值",
      field: "ruleValue",
      required: !0,
      component: "Input"
    },
    {
      label: "状态",
      field: "status",
      required: !0,
      component: "RadioButtonGroup",
      componentProps: {
        options: [
          { label: "有效", value: 1 },
          { label: "无效", value: 0 }
        ]
      },
      defaultValue: 1
    }
  ]) };
}
export {
  a as U,
  n as a,
  i as b,
  u as c,
  r as d,
  c as u
};
