// @jeecg/online@3.6.2-beta/CgformCopyList-14b76108.mjs
import { defineComponent as J, provide as B, watch as k, resolveComponent as e, openBlock as C, createElementBlock as M, Fragment as v, createElementVNode as I, normalizeClass as q, createVNode as t, withCtx as n, createTextVNode as a, createCommentVNode as T } from "vue";
import { BasicTable as L, TableAction as N } from "/@/components/Table";
import { C as V } from "./CgformModal-eccfc38e.mjs";
import F from "./DbToOnlineModal-9f13956f.mjs";
import O from "./CustomButtonList-02e3d897.mjs";
import $ from "./EnhanceJsModal-05c73a59.mjs";
import P from "./EnhanceJavaModal-f63aa2d3.mjs";
import z from "./EnhanceSqlModal-5e4ae84f.mjs";
import Q from "./AuthManagerDrawer-300dd4e4.mjs";
import Y from "./AuthSetterModal-a414fb35.mjs";
import { C as j, u as G } from "./useCgformList-423282b7.mjs";
import { _ as H } from "./index-0e3ab3af.mjs";
import "/@/components/Icon";
import "/@/components/Modal";
import "/@/components/Form/index";
import "/@/hooks/web/useMessage";
import "./useSchemas-0433a403.mjs";
import "ant-design-vue";
import "@ant-design/icons-vue";
import "/@/utils/common/compUtils";
import "/@/hooks/web/usePermission";
import "/@/utils/helper/validator";
import "./DBAttributeTable-370de07b.mjs";
import "/@/components/jeecg/JVxeTable/types";
import "./useTableSync-774c3774.mjs";
import "./cgform.data-cbdd88c3.mjs";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/uuid";
import "lodash-es";
import "./PageAttributeTable-965dc58e.mjs";
import "./LinkTableConfigModal-d4fbdef2.mjs";
import "/@/utils/http/axios";
import "./LinkTableFieldConfigModal-787bc55c.mjs";
import "./FieldExtendJsonModal-a70a9ccd.mjs";
import "./constant-754f1a9d.mjs";
import "./CheckDictTable-134cbb83.mjs";
import "/@/components/jeecg/JPrompt";
import "/@/hooks/web/useDesign";
import "./ForeignKeyTable-6a843c4b.mjs";
import "./IndexTable-f8af9d44.mjs";
import "./QueryTable-eb9a5b21.mjs";
import "./ExtendConfigModal-1dde390a.mjs";
import "/@/components/Form";
import "./useOnlineTest-e4bd8be3.mjs";
import "/@/utils";
import "./useExtendComponent-325fe3f1.mjs";
import "/@/components/Form/src/componentMap";
import "/@/utils/propTypes";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "vue-router";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "/@/components/Loading";
import "/@/utils/auth";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
import "./EnhanceJsHistory-352b46b1.mjs";
import "/@/utils/dateUtil";
import "/@/store";
import "pinia";
import "/@/utils/cache";
import "./enhance.api-138e6826.mjs";
import "/@/utils/is";
import "./enhance.data-efc7e155.mjs";
import "/@/components/Drawer";
import "./AuthFieldConfig-6b21915c.mjs";
import "./auth.api-d3a6c4b3.mjs";
import "./auth.data-8be5c9e3.mjs";
import "./AuthButtonConfig-244ec09f.mjs";
import "./AuthDataConfig-83891eea.mjs";
import "./LeftRole-8b8dfe21.mjs";
import "./LeftDepart-fa85523d.mjs";
import "/@/api/common/api";
import "./LeftUser-dc24ef2f.mjs";
import "./AuthFieldTree-69864163.mjs";
import "./AuthButtonTree-3aa0ed49.mjs";
import "./AuthDataTree-94e026e3.mjs";
import "clipboard";
import "./cgformState-d9f8ec42.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const K = J({
  name: "CgformCopyList",
  components: {
    BasicTable: L,
    TableAction: N,
    CgformModal: V,
    DbToOnlineModal: F,
    CustomButtonList: O,
    EnhanceJsModal: $,
    EnhanceJavaModal: P,
    EnhanceSqlModal: z,
    AuthManagerDrawer: Q,
    AuthSetterModal: Y
  },
  setup() {
    const o = j.copy;
    B("cgformPageType", o);
    const {
      router: S,
      pageContext: E,
      getTableAction: A,
      getDropDownAction: b,
      onShowCustomButton: w,
      onShowEnhanceJs: i,
      onShowEnhanceSql: m,
      onShowEnhanceJava: p,
      registerCustomButtonModal: l,
      registerEnhanceJsModal: s,
      registerEnhanceSqlModal: c,
      registerEnhanceJavaModal: u,
      registerAuthManagerDrawer: g,
      registerAuthSetterModal: d,
      registerCgformModal: h,
      registerDbToOnlineModal: f
    } = G({
      pageType: o,
      designScope: "online-cgform-list",
      columns: [
        { title: "视图表名", dataIndex: "tableName" },
        { title: "视图表描述", dataIndex: "tableTxt" },
        { title: "原表版本", dataIndex: "copyVersion" },
        { title: "视图版本", dataIndex: "tableVersion" }
      ],
      formSchemas: [{ label: "表名", field: "tableName", component: "JInput" }]
    }), { prefixCls: r, tableContext: y } = E, [R, { reload: _ }, { rowSelection: D }] = y;
    return k(S.currentRoute, () => _()), {
      prefixCls: r,
      reload: _,
      rowSelection: D,
      getTableAction: A,
      getDropDownAction: b,
      onShowCustomButton: w,
      onShowEnhanceJs: i,
      onShowEnhanceSql: m,
      onShowEnhanceJava: p,
      registerCustomButtonModal: l,
      registerEnhanceJsModal: s,
      registerEnhanceSqlModal: c,
      registerEnhanceJavaModal: u,
      registerAuthManagerDrawer: g,
      registerAuthSetterModal: d,
      registerTable: R,
      registerCgformModal: h,
      registerDbToOnlineModal: f
    };
  }
}), U = {
  key: 0,
  style: { color: "limegreen" }
}, W = {
  key: 1,
  style: { color: "red" }
};
function X(o, S, E, A, b, w) {
  const i = e("a-button"), m = e("TableAction"), p = e("BasicTable"), l = e("CgformModal"), s = e("EnhanceJsModal"), c = e("EnhanceJavaModal"), u = e("EnhanceSqlModal"), g = e("DbToOnlineModal"), d = e("CustomButtonList"), h = e("AuthManagerDrawer"), f = e("AuthSetterModal");
  return C(), M(v, null, [
    I("div", {
      class: q(o.prefixCls)
    }, [
      t(p, {
        onRegister: o.registerTable,
        rowSelection: o.rowSelection
      }, {
        tableTitle: n(() => [
          t(i, {
            onClick: o.onShowCustomButton,
            type: "primary",
            preIcon: "ant-design:highlight"
          }, {
            default: n(() => [
              a("自定义按钮")
            ]),
            _: 1
          }, 8, ["onClick"]),
          t(i, {
            onClick: o.onShowEnhanceJs,
            type: "primary",
            preIcon: "ant-design:strikethrough"
          }, {
            default: n(() => [
              a("JS增强")
            ]),
            _: 1
          }, 8, ["onClick"]),
          t(i, {
            onClick: o.onShowEnhanceSql,
            type: "primary",
            preIcon: "ant-design:filter"
          }, {
            default: n(() => [
              a("SQL增强")
            ]),
            _: 1
          }, 8, ["onClick"]),
          t(i, {
            onClick: o.onShowEnhanceJava,
            type: "primary",
            preIcon: "ant-design:tool"
          }, {
            default: n(() => [
              a("Java增强")
            ]),
            _: 1
          }, 8, ["onClick"])
        ]),
        dbSync: n(({ text: r }) => [
          r === "Y" ? (C(), M("span", U, "已同步")) : T("", !0),
          r === "N" ? (C(), M("span", W, "未同步")) : T("", !0)
        ]),
        action: n(({ record: r }) => [
          t(m, {
            actions: o.getTableAction(r),
            dropDownActions: o.getDropDownAction(r)
          }, null, 8, ["actions", "dropDownActions"])
        ]),
        _: 1
      }, 8, ["onRegister", "rowSelection"])
    ], 2),
    t(l, {
      onRegister: o.registerCgformModal,
      actionButton: !1,
      onSuccess: o.reload
    }, null, 8, ["onRegister", "onSuccess"]),
    t(s, { onRegister: o.registerEnhanceJsModal }, null, 8, ["onRegister"]),
    t(c, { onRegister: o.registerEnhanceJavaModal }, null, 8, ["onRegister"]),
    t(u, { onRegister: o.registerEnhanceSqlModal }, null, 8, ["onRegister"]),
    t(g, {
      onRegister: o.registerDbToOnlineModal,
      onSuccess: o.reload
    }, null, 8, ["onRegister", "onSuccess"]),
    t(d, { onRegister: o.registerCustomButtonModal }, null, 8, ["onRegister"]),
    t(h, { onRegister: o.registerAuthManagerDrawer }, null, 8, ["onRegister"]),
    t(f, { onRegister: o.registerAuthSetterModal }, null, 8, ["onRegister"])
  ], 64);
}
const qt = /* @__PURE__ */ H(K, [["render", X]]);
export {
  qt as default
};
