// @jeecg/online@3.6.2-beta/CodeFileListModal-a6d3290d.mjs
var M = (d, l, i) => new Promise((e, m) => {
  var p = (o) => {
    try {
      n(i.next(o));
    } catch (r) {
      m(r);
    }
  }, a = (o) => {
    try {
      n(i.throw(o));
    } catch (r) {
      m(r);
    }
  }, n = (o) => o.done ? e(o.value) : Promise.resolve(o.value).then(p, a);
  n((i = i.apply(d, l)).next());
});
import { ref as u, reactive as x, resolveComponent as _, openBlock as C, createElementBlock as h, Fragment as v, createVNode as s, withCtx as c, createTextVNode as f, createElementVNode as y, normalizeStyle as F, renderList as B, toDisplayString as N } from "vue";
import { defHttp as R } from "/@/utils/http/axios";
import "/@/components/Form";
import { BasicModal as T, useModalInner as z, useModal as G } from "/@/components/Modal";
import { InfoCircleTwoTone as K } from "@ant-design/icons-vue";
import S from "./CodeFileViewModal-373e8058.mjs";
import { useMessage as H } from "/@/hooks/web/useMessage";
import { downloadByData as I } from "/@/utils/file/download";
import { _ as E } from "./index-0e3ab3af.mjs";
import "ant-design-vue";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
const j = {
  name: "CodeFileListModal",
  components: {
    BasicModal: T,
    InfoCircleTwoTone: K,
    CodeFileViewModal: S
  },
  emits: ["register"],
  setup() {
    const { createMessage: d } = H(), l = u([]), i = window.innerHeight - 150, e = x({
      overflowY: "auto",
      maxHeight: i + "px"
    }), m = u(!1), p = u(""), a = u(""), [n, { closeModal: o }] = z((t) => M(this, null, function* () {
      l.value = t.codeList, p.value = t.tableName, a.value = t.pathKey;
    }));
    function r() {
      o();
    }
    function g() {
      let t = l.value;
      if (!t || t.length == 0) {
        d.warning("无代码！");
        return;
      }
      let V = t.join(",");
      return R.post(
        {
          url: "/online/cgform/api/downGenerateCode",
          params: {
            fileList: encodeURI(V),
            pathKey: a.value
          },
          responseType: "blob"
        },
        { isTransformResponse: !1 }
      ).then((w) => {
        if (!w || w.size == 0) {
          d.warning("导出代码失败！");
          return;
        }
        let b = "导到生成代码_" + p.value + "_" + (/* @__PURE__ */ new Date()).getTime() + ".zip";
        I(w, b, "application/zip");
      });
    }
    const [L, { openModal: k }] = G();
    function D() {
      let t = l.value;
      k(!0, {
        codeList: t,
        pathKey: a.value
      });
    }
    return {
      registerModal: n,
      registerCodeViewModal: L,
      divStyle: e,
      codeList: l,
      onDownloadGenerateCode: g,
      handleClose: r,
      handleView: D,
      loading: m
    };
  }
}, U = /* @__PURE__ */ y("br", null, null, -1);
function Y(d, l, i, e, m, p) {
  const a = _("info-circle-two-tone"), n = _("a-button"), o = _("BasicModal"), r = _("code-file-view-modal");
  return C(), h(v, null, [
    s(o, {
      onRegister: e.registerModal,
      width: 1200,
      defaultFullscreen: !1,
      canFullscreen: !1
    }, {
      title: c(() => [
        s(a),
        f(" 代码生成结果 ")
      ]),
      footer: c(() => [
        s(n, { onClick: e.handleClose }, {
          default: c(() => [
            f("关闭")
          ]),
          _: 1
        }, 8, ["onClick"]),
        s(n, {
          type: "primary",
          ghost: "",
          onClick: e.handleView
        }, {
          default: c(() => [
            f("在线预览")
          ]),
          _: 1
        }, 8, ["onClick"]),
        s(n, {
          type: "primary",
          onClick: e.onDownloadGenerateCode,
          loading: e.loading
        }, {
          default: c(() => [
            f("下载到本地")
          ]),
          _: 1
        }, 8, ["onClick", "loading"])
      ]),
      default: c(() => [
        y("div", {
          style: F(e.divStyle)
        }, [
          y("p", null, [
            (C(!0), h(v, null, B(e.codeList, (g) => (C(), h(v, null, [
              f(N(g), 1),
              U
            ], 64))), 256))
          ])
        ], 4)
      ]),
      _: 1
    }, 8, ["onRegister"]),
    s(r, {
      onRegister: e.registerCodeViewModal,
      onDownload: e.onDownloadGenerateCode,
      onClose: e.handleClose
    }, null, 8, ["onRegister", "onDownload", "onClose"])
  ], 64);
}
const te = /* @__PURE__ */ E(j, [["render", Y]]);
export {
  te as default
};
