// @jeecg/online@3.6.2-beta/CodeFileViewModal-373e8058.mjs
var k = (l, h, u) => new Promise((w, m) => {
  var v = (i) => {
    try {
      s(u.next(i));
    } catch (r) {
      m(r);
    }
  }, _ = (i) => {
    try {
      s(u.throw(i));
    } catch (r) {
      m(r);
    }
  }, s = (i) => i.done ? w(i.value) : Promise.resolve(i.value).then(v, _);
  s((u = u.apply(l, h)).next());
});
import { defineComponent as O, ref as g, reactive as S, resolveComponent as f, openBlock as x, createBlock as D, withCtx as p, createVNode as C, createTextVNode as R, createElementVNode as W, normalizeStyle as P, createCommentVNode as j } from "vue";
import { defHttp as q } from "/@/utils/http/axios";
import { BasicModal as z, useModalInner as H } from "/@/components/Modal";
import { InfoCircleTwoTone as U } from "@ant-design/icons-vue";
import { message as Y } from "ant-design-vue";
import { JCodeEditor as Q } from "/@/components/Form";
import { _ as X } from "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "/@/hooks/web/useMessage";
import "vue-router";
const Z = O({
  name: "CodeFileViewModal",
  components: {
    BasicModal: z,
    InfoCircleTwoTone: U,
    JCodeEditor: Q
  },
  emits: ["download", "register", "close"],
  setup(l, { emit: h }) {
    const u = g([]), w = g(""), m = g([]), v = g(!1), _ = window.innerHeight - 160, s = g("java"), i = g("");
    let r = S({});
    const [T, { closeModal: B }] = H((t) => k(this, null, function* () {
      r = S({}), i.value = "", u.value = t.codeList, w.value = t.pathKey, y(), v.value = !0;
    }));
    function y() {
      let t = $(), o = t[0];
      E(o, t);
      let n = [];
      const e = function(d) {
        if (d.children) {
          let a = d.children;
          a.length == 1 ? e(a[0]) : a.length > 1 && n.push(d);
        }
      };
      e(o), m.value = n, setTimeout(() => {
        M(o);
      }, 300);
    }
    function M(t) {
      return k(this, null, function* () {
        const o = function(e) {
          if (e.isLeaf === !0)
            return e;
          if (e.children)
            return o(e.children[0]);
        };
        let n = o(t);
        if (n && n.isLeaf === !0) {
          let e = n.path;
          r[e] || (yield K(e)), s.value = A(e), i.value = r[e];
        }
      });
    }
    function E(t, o) {
      for (let n of o)
        t.key == n.pid && (t.children || (t.children = []), t.children.push(n), E(n, o));
    }
    function V(t, o) {
      let n = 0, e = "";
      for (; n <= o; )
        e += t[n], n++;
      return e;
    }
    function $() {
      let t = [], o = [], n = u.value;
      for (let e of n) {
        let d = e.replace(new RegExp("\\\\", "g"), "/").replace("生成成功：", "").trim();
        if (d) {
          let a = d.split("/");
          for (let c = 0; c < a.length; c++) {
            let b = a[c], L = V(a, c);
            if (b) {
              let F = {
                title: b,
                key: L
              };
              if (b != 0) {
                let N = V(a, c - 1);
                N && (F.pid = N);
              }
              c == a.length - 1 && (F.isLeaf = !0, F.path = d), (o.indexOf(L) < 0 || c == a.length - 1) && (t.push(F), o.push(L));
            }
          }
        }
      }
      return t;
    }
    function G() {
      B(), h("close");
    }
    function I() {
      h("download");
    }
    function A(t) {
      return t.endsWith("xml") ? "application/xml" : t.endsWith("sql") ? "text/x-sql" : t.endsWith("vue") ? "text/x-vue" : t.endsWith("ts") ? "text/typescript" : "text/x-java";
    }
    function J(t, o) {
      return k(this, null, function* () {
        let n = o.node.dataRef;
        if (n.isLeaf) {
          let e = n.path;
          r[e] || (yield K(e)), s.value = A(e), i.value = r[e];
        }
      });
    }
    function K(t) {
      return new Promise((o) => {
        let n = {
          path: encodeURI(t),
          pathKey: w.value
        };
        q.get({ url: "/online/cgform/api/codeView", params: n }, { isTransformResponse: !1 }).then((e) => {
          if (!e || e.size === 0) {
            Y.warning("文件下载失败");
            return;
          }
          let d = new Blob([e]), a = new FileReader();
          a.readAsText(d, "utf8"), a.onload = function() {
            let c = this.result;
            r[t] = c, o(1);
          };
        });
      });
    }
    return {
      registerModal: T,
      codeList: u,
      onDownloadGenerateCode: I,
      handleClose: G,
      treeData: m,
      showCodeContent: J,
      activeCodeContent: i,
      expandStatus: v,
      height: _,
      language: s
    };
  }
});
function ee(l, h, u, w, m, v) {
  const _ = f("info-circle-two-tone"), s = f("a-directory-tree"), i = f("a-col"), r = f("JCodeEditor"), T = f("a-empty"), B = f("a-row"), y = f("a-button"), M = f("BasicModal");
  return x(), D(M, {
    onRegister: l.registerModal,
    okText: "",
    cancelText: "关闭",
    width: 1200,
    defaultFullscreen: !0,
    canFullscreen: !1,
    onOk: l.onDownloadGenerateCode
  }, {
    title: p(() => [
      C(_),
      R(" 代码在线预览 ")
    ]),
    footer: p(() => [
      C(y, { onClick: l.handleClose }, {
        default: p(() => [
          R("关闭")
        ]),
        _: 1
      }, 8, ["onClick"]),
      C(y, {
        type: "primary",
        onClick: l.onDownloadGenerateCode
      }, {
        default: p(() => [
          R("下载到本地")
        ]),
        _: 1
      }, 8, ["onClick"])
    ]),
    default: p(() => [
      W("div", null, [
        C(B, null, {
          default: p(() => [
            C(i, {
              span: 6,
              gutter: 3,
              style: { "border-right": "1px solid #eee" }
            }, {
              default: p(() => [
                W("div", {
                  style: P({ height: l.height + "px", overflowY: "auto" })
                }, [
                  l.treeData.length ? (x(), D(s, {
                    key: 0,
                    defaultExpandAll: !0,
                    "tree-data": l.treeData,
                    onSelect: l.showCodeContent
                  }, null, 8, ["tree-data", "onSelect"])) : j("", !0)
                ], 4)
              ]),
              _: 1
            }),
            C(i, {
              span: 18,
              gutter: 3
            }, {
              default: p(() => [
                l.activeCodeContent ? (x(), D(r, {
                  key: 0,
                  value: l.activeCodeContent,
                  "onUpdate:value": h[0] || (h[0] = (E) => l.activeCodeContent = E),
                  theme: "idea",
                  language: l.language,
                  fullScreen: !1,
                  lineNumbers: !0,
                  height: l.height + "px",
                  "language-change": !0
                }, null, 8, ["value", "language", "height"])) : (x(), D(T, {
                  key: 1,
                  style: { "margin-top": "50px" },
                  description: "请选择左侧文件，显示详细代码"
                }))
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ])
    ]),
    _: 1
  }, 8, ["onRegister", "onOk"]);
}
const fe = /* @__PURE__ */ X(Z, [["render", ee]]);
export {
  fe as default
};
