// @jeecg/online@3.6.2-beta/EnhanceJsHistory-352b46b1.mjs
var x = Object.defineProperty;
var b = Object.getOwnPropertySymbols;
var N = Object.prototype.hasOwnProperty, $ = Object.prototype.propertyIsEnumerable;
var S = (e, n, t) => n in e ? x(e, n, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[n] = t, v = (e, n) => {
  for (var t in n || (n = {}))
    N.call(n, t) && S(e, t, n[t]);
  if (b)
    for (var t of b(n))
      $.call(n, t) && S(e, t, n[t]);
  return e;
};
var M = (e, n, t) => new Promise((u, _) => {
  var m = (s) => {
    try {
      r(t.next(s));
    } catch (d) {
      _(d);
    }
  }, p = (s) => {
    try {
      r(t.throw(s));
    } catch (d) {
      _(d);
    }
  }, r = (s) => s.done ? u(s.value) : Promise.resolve(s.value).then(m, p);
  r((t = t.apply(e, n)).next());
});
import { formatToDate as j } from "/@/utils/dateUtil";
import { defineComponent as I, ref as g, nextTick as R, resolveComponent as c, openBlock as w, createBlock as T, withCtx as a, createVNode as i, createTextVNode as k, normalizeClass as D, createElementVNode as F, toDisplayString as V } from "vue";
import { BasicModal as A, useModalInner as O } from "/@/components/Modal";
import { JCodeEditor as z } from "/@/components/Form";
import "/@/store";
import { defineStore as P } from "pinia";
import { createLocalStorage as q } from "/@/utils/cache";
import { _ as G } from "./index-0e3ab3af.mjs";
const B = q(), H = "enhance_", K = P({
  id: "online-cgform-enhance",
  state: () => ({
    enhanceJs: {}
  }),
  getters: {},
  actions: {
    getEnhanceJs(e) {
      return this.enhanceJs[e] = B.get(H + e), this.enhanceJs[e];
    },
    addEnhanceJs(e) {
      this.enhanceJs[e.code] ? this.enhanceJs[e.code].push(v({}, e)) : this.enhanceJs[e.code] = [v({}, e)];
      let n = this.enhanceJs[e.code];
      for (; n.length > 16; )
        n.shift();
      B.set(H + e.code, n);
    }
  }
}), Q = I({
  name: "EnhanceJsHistory",
  components: { BasicModal: A, JCodeEditor: z },
  setup() {
    const e = K(), n = g(), t = g(!1), u = g([]), _ = g(""), m = g(0), [p, { closeModal: r }] = O((o) => M(this, null, function* () {
      s(o.code, o.type);
    }));
    function s(o, E) {
      _.value = "", u.value = [];
      let J = e.getEnhanceJs(o), l = [], f = 0;
      for (let h of J)
        h.type === E && (f++, l.push(Object.assign({}, h, { index: f })));
      l && l.length > 0 && l.sort((h, L) => L.date - h.date), u.value = [...l], R(() => y(l[0]));
    }
    function d() {
      r();
    }
    function C(o) {
      return j(o, "yyyy-MM-DD HH:mm:ss");
    }
    function y(o) {
      m.value = o.index, n.value.setValue(o.str);
    }
    return {
      codeEditorRef: n,
      fullCode: y,
      registerModal: p,
      getFormatDate: C,
      onCancel: d,
      confirmLoading: t,
      dataList: u,
      jsStr: _,
      activeIndex: m
    };
  }
});
const U = ["onClick"];
function W(e, n, t, u, _, m) {
  const p = c("a-divider"), r = c("a-list-item"), s = c("a-list"), d = c("a-layout-sider"), C = c("JCodeEditor"), y = c("a-layout-content"), o = c("a-layout"), E = c("a-spin"), J = c("a-button"), l = c("BasicModal");
  return w(), T(l, {
    onRegister: e.registerModal,
    title: "JS增强历史记录",
    width: 1200,
    maskClosable: !1,
    confirmLoading: e.confirmLoading,
    defaultFullscreen: "",
    onCancel: e.onCancel
  }, {
    footer: a(() => [
      i(J, { onClick: e.onCancel }, {
        default: a(() => [
          k("关闭")
        ]),
        _: 1
      }, 8, ["onClick"])
    ]),
    default: a(() => [
      i(E, { spinning: e.confirmLoading }, {
        default: a(() => [
          i(o, null, {
            default: a(() => [
              i(d, { theme: "light" }, {
                default: a(() => [
                  i(s, {
                    bordered: "",
                    dataSource: e.dataList,
                    class: D("enhance-list")
                  }, {
                    header: a(() => [
                      F("div", null, [
                        i(p, { style: { margin: "0" } }, {
                          default: a(() => [
                            k("保存时间")
                          ]),
                          _: 1
                        })
                      ])
                    ]),
                    renderItem: a(({ item: f }) => [
                      i(r, {
                        class: D(e.activeIndex === f.index ? "bg-blue" : "")
                      }, {
                        default: a(() => [
                          F("a", {
                            onClick: (h) => e.fullCode(f)
                          }, V(e.getFormatDate(f.date)), 9, U)
                        ]),
                        _: 2
                      }, 1032, ["class"])
                    ]),
                    _: 1
                  }, 8, ["dataSource"])
                ]),
                _: 1
              }),
              i(o, null, {
                default: a(() => [
                  i(y, { style: { margin: "8px 8px", padding: "8px", background: "#fff", minHeight: "280px" } }, {
                    default: a(() => [
                      i(C, {
                        ref: "codeEditorRef",
                        language: "javascript",
                        fullScreen: !0,
                        lineNumbers: !1,
                        "language-change": !1
                      }, null, 512)
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["spinning"])
    ]),
    _: 1
  }, 8, ["onRegister", "confirmLoading", "onCancel"]);
}
const X = /* @__PURE__ */ G(Q, [["render", W], ["__scopeId", "data-v-b15e9c26"]]), ie = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: X
}, Symbol.toStringTag, { value: "Module" }));
export {
  X as E,
  ie as a,
  K as u
};
