// @jeecg/online@3.6.2-beta/IndexTable-f8af9d44.mjs
import { defineComponent as f, ref as b, resolveComponent as g, openBlock as h, createBlock as y, mergeProps as $ } from "vue";
import { JVxeTypes as a } from "/@/components/jeecg/JVxeTable/types";
import { u as T } from "./useTableSync-774c3774.mjs";
import { _ as v } from "./index-0e3ab3af.mjs";
import "./cgform.data-cbdd88c3.mjs";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/uuid";
import "lodash-es";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "/@/hooks/web/useMessage";
import "vue-router";
const S = f({
  name: "IndexTable",
  components: {},
  props: {
    actionButton: {
      type: Boolean,
      default: !0,
      required: !1
    }
  },
  setup() {
    const e = b([
      {
        title: "索引名称",
        key: "indexName",
        width: 330,
        type: a.input,
        defaultValue: "",
        placeholder: "请输入${title}",
        validateRules: [{ required: !0, message: "${title}不能为空" }]
      },
      {
        title: "索引栏位",
        key: "indexField",
        width: 330,
        type: a.selectMultiple,
        options: [],
        defaultValue: "",
        placeholder: "请选择${title}",
        validateRules: [{ required: !0, message: "请选择${title}" }]
      },
      {
        title: "索引类型",
        key: "indexType",
        width: 330,
        type: a.select,
        options: [
          { title: "normal", value: "normal" },
          { title: "unique", value: "unique" }
        ],
        defaultValue: "normal",
        placeholder: "请选择${title}",
        validateRules: [{ required: !0, message: "请选择${title}" }]
      }
    ]), l = T(e), { tableRef: r, loading: i, dataSource: n, tableHeight: u, tableProps: t, setDataSource: s, validateData: p } = l;
    function m(c) {
      let d = [];
      c.value.tableRef.getTableData().forEach((o) => {
        o.dbFieldName && d.push({
          title: o.dbFieldName,
          value: o.dbFieldName
        });
      }), e.value[1].options = d;
    }
    return { tableRef: r, loading: i, dataSource: n, columns: e, tableHeight: u, tableProps: t, syncTable: m, setDataSource: s, validateData: p };
  }
});
function x(e, l, r, i, n, u) {
  const t = g("JVxeTable");
  return h(), y(t, $({
    ref: "tableRef",
    rowNumber: "",
    rowSelection: "",
    dragSort: "",
    keyboardEdit: "",
    sortKey: "orderNum",
    maxHeight: e.tableHeight.normal,
    loading: e.loading,
    columns: e.columns,
    dataSource: e.dataSource,
    toolbar: e.actionButton
  }, e.tableProps), null, 16, ["maxHeight", "loading", "columns", "dataSource", "toolbar"]);
}
const C = /* @__PURE__ */ v(S, [["render", x]]);
export {
  C as default
};
