// @jeecg/online@3.6.2-beta/JModalTip-33c7dc44.mjs
import { computed as c, openBlock as n, createElementBlock as l, normalizeClass as p, createElementVNode as t, createCommentVNode as d, pushScopeId as r, popScopeId as _ } from "vue";
import { useAppInject as m } from "/@/hooks/web/useAppInject";
import { _ as v } from "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "/@/hooks/web/useMessage";
import "vue-router";
const f = {
  name: "JModalTip",
  props: {
    visible: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["save", "cancel"],
  setup(e) {
    const { getIsMobile: o } = m();
    return {
      flag: c(() => e.visible),
      getIsMobile: o
    };
  }
}, u = (e) => (r("data-v-dc9246f4"), e = e(), _(), e), g = { class: "container" }, I = { class: "outer" }, b = { class: "inner" }, $ = /* @__PURE__ */ u(() => /* @__PURE__ */ t("span", { class: "tip" }, "正在修改表单数据 ···", -1));
function k(e, o, i, s, M, y) {
  return s.flag ? (n(), l("div", {
    key: 0,
    class: p(["jeecg-update-tip-bar", { mobile: s.getIsMobile }])
  }, [
    t("div", g, [
      t("div", I, [
        t("div", b, [
          $,
          t("div", {
            class: "cancel",
            onClick: o[0] || (o[0] = (a) => e.$emit("cancel"))
          }, "取消"),
          t("div", {
            class: "save",
            onClick: o[1] || (o[1] = (a) => e.$emit("save"))
          }, "保存")
        ])
      ])
    ])
  ], 2)) : d("", !0);
}
const j = /* @__PURE__ */ v(f, [["render", k], ["__scopeId", "data-v-dc9246f4"]]);
export {
  j as default
};
