// @jeecg/online@3.6.2-beta/LeftRole-8b8dfe21.mjs
var m = (r, l, o) => new Promise((i, n) => {
  var c = (e) => {
    try {
      a(o.next(e));
    } catch (t) {
      n(t);
    }
  }, s = (e) => {
    try {
      a(o.throw(e));
    } catch (t) {
      n(t);
    }
  }, a = (e) => e.done ? i(e.value) : Promise.resolve(e.value).then(c, s);
  a((o = o.apply(r, l)).next());
});
import { defineComponent as g, resolveComponent as _, openBlock as w, createBlock as b, withCtx as S, createElementVNode as T } from "vue";
import { defHttp as h } from "/@/utils/http/axios";
import { BasicTable as C } from "/@/components/Table";
import { useListPage as R } from "/@/hooks/system/useListPage";
import { _ as x } from "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "/@/hooks/web/useMessage";
import "vue-router";
const B = g({
  name: "LeftRole",
  components: { BasicTable: C },
  emits: ["select"],
  setup(r, { emit: l }) {
    const { tableContext: o, createMessage: i } = R({
      tableProps: {
        api: a,
        rowKey: "id",
        size: "small",
        bordered: !0,
        columns: [
          { title: "角色编码", align: "center", dataIndex: "roleCode" },
          { title: "角色名称", align: "center", dataIndex: "roleName" }
        ],
        rowSelection: {
          type: "radio",
          onChange(t) {
            t.length > 0 && l("select", t[0]);
          }
        },
        canResize: !1,
        clickToRowSelect: !0,
        useSearchForm: !1,
        showActionColumn: !1,
        showTableSetting: !1
      }
    }), [n, { clearSelectedRowKeys: c }, { rowSelection: s }] = o;
    function a(t) {
      return m(this, null, function* () {
        let { code: p, success: f, result: u, message: d } = yield h.get(
          {
            url: "/sys/role/list",
            params: t
          },
          { isTransformResponse: !1 }
        );
        return f ? u : (p === 510 && i.warning(d), {});
      });
    }
    function e() {
      c();
    }
    return { rowSelection: s, registerTable: n, clearSelected: e };
  }
}), $ = /* @__PURE__ */ T("span", null, null, -1);
function y(r, l, o, i, n, c) {
  const s = _("BasicTable");
  return w(), b(s, {
    onRegister: r.registerTable,
    rowSelection: r.rowSelection
  }, {
    tableTop: S(() => [
      $
    ]),
    _: 1
  }, 8, ["onRegister", "rowSelection"]);
}
const E = /* @__PURE__ */ x(B, [["render", y]]);
export {
  E as default
};
