// @jeecg/online@3.6.2-beta/OnlCgformErpList-db45e22b.mjs
var T = (K, _, d) => new Promise((b, S) => {
  var w = (i) => {
    try {
      u(d.next(i));
    } catch (h) {
      S(h);
    }
  }, L = (i) => {
    try {
      u(d.throw(i));
    } catch (h) {
      S(h);
    }
  }, u = (i) => i.done ? b(i.value) : Promise.resolve(i.value).then(w, L);
  u((d = d.apply(K, _)).next());
});
import { ref as k, watch as $, resolveComponent as I, openBlock as l, createElementBlock as m, createElementVNode as E, createVNode as c, unref as e, createBlock as s, normalizeClass as Ne, withCtx as r, createTextVNode as g, createCommentVNode as p, Fragment as F, renderList as j, toDisplayString as M, withDirectives as $e, vShow as je, mergeProps as Ve, toHandlers as qe, resolveDynamicComponent as Ue, normalizeProps as Ge, guardReactiveProps as We } from "vue";
import { BasicTable as Xe, TableAction as Ye } from "/@/components/Table";
import { useMessage as Ze } from "/@/hooks/web/useMessage";
import eo from "./OnlineAutoModal-510991eb.mjs";
import oo from "./OnlineCustomModal-518ad33c.mjs";
import V from "./OnlineDetailModal-9675390d.mjs";
import to from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as no, a as io } from "./useListButton-cdbab702.mjs";
import { u as lo, a as ro, O as ao, g as so } from "./useExtendComponent-325fe3f1.mjs";
import co from "./OnlineQueryForm-1bfc8320.mjs";
import uo from "./SuperQuery-5d1ca957.mjs";
import { u as po } from "./useOnlinePopEvent-687070b7.mjs";
import mo from "./OnlCgformErpSubTable-54b50eec.mjs";
import { E as q } from "./constant-754f1a9d.mjs";
import { cloneDeep as go } from "lodash-es";
import "/@/components/Modal";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "/@/utils/http/axios";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "./OnlineFormDetail-0e7294ca.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
import "./SuperQueryValComponent.vue_vue_type_script_lang-e9196096.mjs";
import "/@/utils/is";
const fo = { class: "p-2 cgformErpList" }, yo = { class: "content" }, _o = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, bo = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, ho = ["src", "onClick"], Co = ["innerHTML"], vo = ["title"], Jt = {
  __name: "OnlCgformErpList",
  setup(K) {
    const _ = k([]), d = k(0), b = k(null), { createMessage: S } = Ze(), w = k({}), L = (t, n) => {
      w.value[t] = n;
    }, {
      ID: u,
      onlineTableContext: i,
      onlineQueryFormOuter: h,
      loading: O,
      reload: f,
      dataSource: Q,
      pagination: U,
      handleSpecialConfig: G,
      getColumnList: W,
      handleChangeInTable: X,
      loadData: Y,
      superQueryButtonRef: x,
      superQueryStatus: Z,
      handleSuperQuery: ee,
      onlineExtConfigJson: J,
      handleFormConfig: oe,
      registerCustomModal: te,
      tableReloading: ne
    } = no({ themeTemplate: q });
    if (!u.value)
      throw S.warning("地址错误, 配置ID不存在!"), new Error("地址错误, 配置ID不存在!");
    let { initCgEnhanceJs: ie } = lo(i);
    const {
      buttonSwitch: C,
      cgLinkButtonList: ko,
      cgTopButtonList: B,
      importUrl: le,
      registerModal: re,
      handleAdd: ae,
      handleEdit: So,
      handleBatchDelete: se,
      registerImportModal: ce,
      onImportExcel: ue,
      onExportExcel: pe,
      cgButtonJsHandler: de,
      cgButtonActionHandler: me,
      cgButtonLinkHandler: wo,
      handleSubmitFlow: Ro,
      getDropDownActions: ge,
      getActions: fe,
      initButtonList: ye,
      initButtonSwitch: _e,
      registerDetailModal: be,
      registerBpmModal: To
    } = io(i, J, {
      singleDelCallback: (t) => {
        Array.isArray(i.selectedRowKeys) && i.selectedRowKeys.includes(t) && i.clearSelectedRow();
      },
      editClickCallback: (t, n) => {
        Array.isArray(i.selectedRowKeys) && i.selectedRowKeys.includes(t) && n.stopPropagation();
      }
    }), D = k(!1);
    function he() {
      return T(this, null, function* () {
        try {
          D.value = !0, yield pe();
        } finally {
          setTimeout(() => D.value = !1, 1500);
        }
      });
    }
    const {
      columns: Ce,
      actionColumn: ve,
      selectedKeys: v,
      rowSelection: ke,
      enableScrollBar: Se,
      tableScroll: we,
      downloadRowFile: Re,
      getImgView: Te,
      getPcaText: H,
      getFormatDate: Ie,
      handleColumnResult: Ee,
      hrefComponent: R,
      viewOnlineCellImage: Me,
      hrefMainTableId: Le,
      registerOnlineHrefModal: Oe,
      registerPopModal: xe,
      openPopModal: Be,
      onlinePopModalRef: De,
      popTableId: z
    } = ro(i, J);
    $(
      v,
      (t) => {
        var n;
        (n = v.value) != null && n.length ? b.value = Q.value.find((a) => a.id === v.value[0]) : b.value = null;
      },
      {
        immediate: !0
      }
    ), $(
      u,
      () => {
        Pe();
      },
      { immediate: !0 }
    );
    function Pe() {
      return T(this, null, function* () {
        O.value = !0;
        let t = yield W(q);
        Ae(t.main), _.value = t.subList, yield Y(), O.value = !1, i.execButtonEnhance("setup");
      });
    }
    function Ae(t) {
      let n = ie(t.enhanceJs);
      i.EnhanceJS = n, ye(t.cgButtonList), _e(t.hideColumns), Ee(t, "radio"), G(t);
    }
    function Fe(t) {
      i.queryParam = t, f({ mode: "search" });
    }
    function Ke() {
      return T(this, arguments, function* (t = {}) {
        yield so(x);
        const n = go(t), { properties: a = {} } = n;
        Object.entries(a).forEach(([P, A]) => {
          A.view == "table" && delete a[P];
        }), x.value.init(n);
      });
    }
    function Qe(t) {
      z.value = t.id;
      let n = {
        title: t.describe
      };
      t.record && t.record.id && (n.record = t.record, n.isUpdate = !0), Be(!0, n);
    }
    po(Qe);
    const Je = (t, n) => {
      ee(t, n), v.value = [];
    };
    return (t, n) => {
      var N;
      const a = I("a-button"), P = I("a-tab-pane"), A = I("a-tabs"), He = I("a-modal");
      return l(), m("div", fo, [
        E("div", yo, [
          c(co, {
            ref_key: "onlineQueryFormOuter",
            ref: h,
            id: e(u),
            onSearch: Fe,
            onLoaded: Ke
          }, null, 8, ["id"]),
          e(ne) ? p("", !0) : (l(), s(e(Xe), {
            key: 0,
            ref: "onlineTable",
            rowKey: "jeecg_row_key",
            canResize: !0,
            bordered: !0,
            showIndexColumn: !1,
            loading: e(O),
            columns: e(Ce),
            dataSource: e(Q),
            pagination: e(U),
            rowSelection: e(ke),
            actionColumn: e(ve),
            showTableSetting: !0,
            clickToRowSelect: !0,
            scroll: e(we),
            onTableRedo: e(f),
            class: Ne({ "j-table-force-nowrap": e(Se) }),
            onChange: e(X)
          }, {
            tableTitle: r(() => [
              e(C).add ? (l(), s(a, {
                key: 0,
                preIcon: "ant-design:plus-outlined",
                type: "primary",
                onClick: e(ae)
              }, {
                default: r(() => [
                  g("新增")
                ]),
                _: 1
              }, 8, ["onClick"])) : p("", !0),
              e(C).import ? (l(), s(a, {
                key: 1,
                preIcon: "ant-design:import-outlined",
                type: "primary",
                onClick: e(ue)
              }, {
                default: r(() => [
                  g("导入")
                ]),
                _: 1
              }, 8, ["onClick"])) : p("", !0),
              e(C).export ? (l(), s(a, {
                key: 2,
                preIcon: "ant-design:export-outlined",
                type: "primary",
                loading: D.value,
                onClick: he
              }, {
                default: r(() => [
                  g(" 导出")
                ]),
                _: 1
              }, 8, ["loading"])) : p("", !0),
              e(B) && e(B).length > 0 ? (l(!0), m(F, { key: 3 }, j(e(B), (o, y) => (l(), m(F, null, [
                o.optType == "js" ? (l(), s(a, {
                  key: "cgbtn" + y,
                  onClick: (ze) => e(de)(o.buttonCode),
                  type: "primary",
                  preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
                }, {
                  default: r(() => [
                    g(M(o.buttonName), 1)
                  ]),
                  _: 2
                }, 1032, ["onClick", "preIcon"])) : o.optType == "action" ? (l(), s(a, {
                  key: "cgbtn" + y,
                  onClick: (ze) => e(me)(o.buttonCode),
                  type: "primary",
                  preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
                }, {
                  default: r(() => [
                    g(M(o.buttonName), 1)
                  ]),
                  _: 2
                }, 1032, ["onClick", "preIcon"])) : p("", !0)
              ], 64))), 256)) : p("", !0),
              e(C).batch_delete ? $e((l(), s(a, {
                key: 4,
                preIcon: "ant-design:delete-outlined",
                onClick: e(se)
              }, {
                default: r(() => [
                  g(" 批量删除")
                ]),
                _: 1
              }, 8, ["onClick"])), [
                [je, e(v).length > 0]
              ]) : p("", !0),
              e(C).super_query ? (l(), s(uo, {
                key: 5,
                ref_key: "superQueryButtonRef",
                ref: x,
                online: "",
                status: e(Z),
                onSearch: Je
              }, null, 8, ["status"])) : p("", !0)
            ]),
            fileSlot: r(({ text: o }) => [
              o ? (l(), s(a, {
                key: 1,
                ghost: !0,
                type: "primary",
                preIcon: "ant-design:download",
                size: "small",
                onClick: (y) => e(Re)(o)
              }, {
                default: r(() => [
                  g(" 下载 ")
                ]),
                _: 2
              }, 1032, ["onClick"])) : (l(), m("span", _o, "无文件"))
            ]),
            imgSlot: r(({ text: o }) => [
              o ? (l(), m("img", {
                key: 1,
                src: e(Te)(o),
                alt: "图片不存在",
                class: "online-cell-image",
                onClick: (y) => e(Me)(o)
              }, null, 8, ho)) : (l(), m("span", bo, "无图片"))
            ]),
            htmlSlot: r(({ text: o }) => [
              E("div", { innerHTML: o }, null, 8, Co)
            ]),
            pcaSlot: r(({ text: o }) => [
              E("div", {
                title: e(H)(o)
              }, M(e(H)(o)), 9, vo)
            ]),
            dateSlot: r(({ text: o }) => [
              E("span", null, M(e(Ie)(o)), 1)
            ]),
            action: r(({ record: o }) => [
              c(e(Ye), {
                actions: e(fe)(o),
                dropDownActions: e(ge)(o)
              }, null, 8, ["actions", "dropDownActions"])
            ]),
            _: 1
          }, 8, ["loading", "columns", "dataSource", "pagination", "rowSelection", "actionColumn", "scroll", "onTableRedo", "class", "onChange"])),
          (N = _.value) != null && N.length ? (l(), s(A, {
            key: 1,
            animated: "",
            activeKey: d.value,
            "onUpdate:activeKey": n[0] || (n[0] = (o) => d.value = o),
            style: { margin: "10px" }
          }, {
            default: r(() => [
              (l(!0), m(F, null, j(_.value, (o, y) => (l(), s(P, {
                tab: o.description,
                key: y,
                forceRender: ""
              }, {
                default: r(() => [
                  c(mo, {
                    data: o,
                    mainTableSelectedRowRcord: b.value,
                    onGetSource: L
                  }, null, 8, ["data", "mainTableSelectedRowRcord"])
                ]),
                _: 2
              }, 1032, ["tab"]))), 128))
            ]),
            _: 1
          }, 8, ["activeKey"])) : p("", !0),
          c(eo, {
            id: e(u),
            subTableSource: w.value,
            onRegister: e(re),
            onSuccess: e(f),
            onFormConfig: e(oe)
          }, null, 8, ["id", "subTableSource", "onRegister", "onSuccess", "onFormConfig"]),
          c(V, {
            id: e(u),
            onRegister: e(be)
          }, null, 8, ["id", "onRegister"]),
          c(to, {
            onRegister: e(ce),
            url: e(le)(),
            onOk: e(f),
            online: ""
          }, null, 8, ["onRegister", "url", "onOk"]),
          c(He, Ve(e(R).model, qe(e(R).on)), {
            default: r(() => [
              (l(), s(Ue(e(R).is), Ge(We(e(R).params)), null, 16))
            ]),
            _: 1
          }, 16),
          c(oo, {
            onRegister: e(te),
            onSuccess: e(f)
          }, null, 8, ["onRegister", "onSuccess"]),
          c(V, {
            id: e(Le),
            onRegister: e(Oe),
            defaultFullscreen: !1
          }, null, 8, ["id", "onRegister"]),
          c(ao, {
            ref_key: "onlinePopModalRef",
            ref: De,
            id: e(z),
            onRegister: e(xe),
            onSuccess: e(f),
            request: "",
            topTip: ""
          }, null, 8, ["id", "onRegister", "onSuccess"])
        ])
      ]);
    };
  }
};
export {
  Jt as default
};
