// @jeecg/online@3.6.2-beta/OnlCgformInnerTableList-782eaf04.mjs
var v = (P, c, l) => new Promise((x, I) => {
  var b = (s) => {
    try {
      h(l.next(s));
    } catch (u) {
      I(u);
    }
  }, g = (s) => {
    try {
      h(l.throw(s));
    } catch (u) {
      I(u);
    }
  }, h = (s) => s.done ? x(s.value) : Promise.resolve(s.value).then(b, g);
  h((l = l.apply(P, c)).next());
});
import { reactive as Je, ref as L, watch as ze, resolveComponent as w, openBlock as n, createElementBlock as y, createVNode as p, unref as e, createBlock as a, normalizeClass as He, withCtx as i, createTextVNode as _, createCommentVNode as d, Fragment as D, renderList as K, toDisplayString as R, withDirectives as $e, vShow as je, createElementVNode as N, mergeProps as Ve, toHandlers as qe, resolveDynamicComponent as Ue, normalizeProps as We, guardReactiveProps as Ge, nextTick as Xe } from "vue";
import { BasicTable as Ye, TableAction as Ze } from "/@/components/Table";
import { useMessage as eo } from "/@/hooks/web/useMessage";
import oo from "./OnlineAutoModal-510991eb.mjs";
import to from "./OnlineCustomModal-518ad33c.mjs";
import J from "./OnlineDetailModal-9675390d.mjs";
import no from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as io, a as ro } from "./useListButton-cdbab702.mjs";
import { u as ao, a as lo, O as so, g as co } from "./useExtendComponent-325fe3f1.mjs";
import uo from "./OnlineQueryForm-1bfc8320.mjs";
import po from "./SuperQuery-5d1ca957.mjs";
import { u as mo } from "./useOnlinePopEvent-687070b7.mjs";
import go from "./OnlCgformInnerSubTable-fbca73e0.mjs";
import { I as bo } from "./constant-754f1a9d.mjs";
import "/@/components/Modal";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "/@/utils/http/axios";
import "lodash-es";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "./OnlineFormDetail-0e7294ca.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
import "./SuperQueryValComponent.vue_vue_type_script_lang-e9196096.mjs";
import "/@/utils/is";
import "/@/components/Table/src/const";
const fo = { class: "p-2" }, yo = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, _o = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, ho = ["src", "onClick"], Co = ["innerHTML"], ko = ["title"], Ft = {
  __name: "OnlCgformInnerTableList",
  setup(P) {
    const c = Je({
      tabNav: [],
      tabIndex: "0"
    }), l = L([]), x = L(null), { createMessage: I } = eo(), {
      ID: b,
      onlineTableContext: g,
      onlineQueryFormOuter: h,
      loading: s,
      reload: u,
      dataSource: z,
      pagination: H,
      handleSpecialConfig: $,
      getColumnList: j,
      handleChangeInTable: V,
      loadData: q,
      superQueryButtonRef: M,
      superQueryStatus: U,
      handleSuperQuery: W,
      onlineExtConfigJson: F,
      handleFormConfig: G,
      registerCustomModal: X,
      tableReloading: Y
    } = io();
    if (!b.value)
      throw I.warning("地址错误, 配置ID不存在!"), new Error("地址错误, 配置ID不存在!");
    let { initCgEnhanceJs: Z } = ao(g);
    const {
      buttonSwitch: C,
      cgTopButtonList: O,
      importUrl: ee,
      registerModal: oe,
      handleAdd: te,
      handleBatchDelete: ne,
      registerImportModal: ie,
      onImportExcel: re,
      onExportExcel: ae,
      cgButtonJsHandler: le,
      cgButtonActionHandler: se,
      getDropDownActions: ce,
      getActions: ue,
      initButtonList: pe,
      initButtonSwitch: de,
      registerDetailModal: me,
      registerBpmModal: Io
    } = ro(g, F), E = L(!1);
    function ge() {
      return v(this, null, function* () {
        try {
          E.value = !0, yield ae();
        } finally {
          setTimeout(() => E.value = !1, 1500);
        }
      });
    }
    const {
      columns: be,
      actionColumn: fe,
      selectedKeys: ye,
      rowSelection: _e,
      enableScrollBar: he,
      tableScroll: Ce,
      downloadRowFile: ke,
      getImgView: Ie,
      getPcaText: A,
      getFormatDate: Se,
      handleColumnResult: Te,
      hrefComponent: S,
      viewOnlineCellImage: ve,
      hrefMainTableId: we,
      registerOnlineHrefModal: Re,
      registerPopModal: xe,
      openPopModal: Me,
      onlinePopModalRef: Oe,
      popTableId: Q
    } = lo(g, F);
    ze(
      b,
      () => {
        Ee();
      },
      { immediate: !0 }
    );
    function Ee() {
      return v(this, null, function* () {
        s.value = !0;
        let t = yield j(bo);
        Be(t), yield q(), s.value = !1, g.execButtonEnhance("setup");
      });
    }
    function Be(t) {
      let r = Z(t.enhanceJs);
      g.EnhanceJS = r, pe(t.cgButtonList), de(t.hideColumns), Te(t), $(t);
    }
    function Le(t) {
      g.queryParam = t, u({ mode: "search" });
    }
    function De(t) {
      return v(this, null, function* () {
        yield co(M), M.value.init(t);
      });
    }
    function Ne(t) {
      Q.value = t.id;
      let r = {
        title: t.describe
      };
      t.record && t.record.id && (r.record = t.record, r.isUpdate = !0), Me(!0, r);
    }
    mo(Ne);
    const Pe = (t) => {
      G(t);
      const { schema: r } = t, { properties: m } = r, T = [];
      Object.entries(m).forEach(([B, k]) => {
        k.view == "tab" && T.push({ tableName: B, tableTxt: k.describe, id: k.id });
      }), c.tabNav = T;
    }, Fe = (t, r) => {
      l.value = [], t && (l.value = [r.id], x.value = r);
    }, Ae = (t) => {
      u(t), Qe();
    }, Qe = () => {
      if (l.value.length) {
        const t = c.tabIndex;
        c.tabIndex = "-1", Xe(() => {
          c.tabIndex = t;
        });
      }
    };
    return (t, r) => {
      const m = w("a-button"), T = w("a-tab-pane"), B = w("a-tabs"), k = w("a-modal");
      return n(), y("div", fo, [
        p(uo, {
          ref_key: "onlineQueryFormOuter",
          ref: h,
          id: e(b),
          onSearch: Le,
          onLoaded: De
        }, null, 8, ["id"]),
        e(Y) ? d("", !0) : (n(), a(e(Ye), {
          key: 0,
          ref: "onlineTable",
          rowKey: "jeecg_row_key",
          canResize: !0,
          bordered: !0,
          showIndexColumn: !1,
          loading: e(s),
          columns: e(be),
          dataSource: e(z),
          pagination: e(H),
          rowSelection: e(_e),
          actionColumn: e(fe),
          showTableSetting: !0,
          clickToRowSelect: !1,
          scroll: e(Ce),
          onTableRedo: e(u),
          class: He({ "j-table-force-nowrap": e(he) }),
          onChange: e(V),
          expandedRowKeys: l.value,
          onExpand: Fe
        }, {
          tableTitle: i(() => [
            e(C).add ? (n(), a(m, {
              key: 0,
              preIcon: "ant-design:plus-outlined",
              type: "primary",
              onClick: e(te)
            }, {
              default: i(() => [
                _("新增")
              ]),
              _: 1
            }, 8, ["onClick"])) : d("", !0),
            e(C).import ? (n(), a(m, {
              key: 1,
              preIcon: "ant-design:import-outlined",
              type: "primary",
              onClick: e(re)
            }, {
              default: i(() => [
                _("导入")
              ]),
              _: 1
            }, 8, ["onClick"])) : d("", !0),
            e(C).export ? (n(), a(m, {
              key: 2,
              preIcon: "ant-design:export-outlined",
              type: "primary",
              loading: E.value,
              onClick: ge
            }, {
              default: i(() => [
                _(" 导出")
              ]),
              _: 1
            }, 8, ["loading"])) : d("", !0),
            e(O) && e(O).length > 0 ? (n(!0), y(D, { key: 3 }, K(e(O), (o, f) => (n(), y(D, null, [
              o.optType == "js" ? (n(), a(m, {
                key: "cgbtn" + f,
                onClick: (Ke) => e(le)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: i(() => [
                  _(R(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : o.optType == "action" ? (n(), a(m, {
                key: "cgbtn" + f,
                onClick: (Ke) => e(se)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: i(() => [
                  _(R(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : d("", !0)
            ], 64))), 256)) : d("", !0),
            e(C).batch_delete ? $e((n(), a(m, {
              key: 4,
              preIcon: "ant-design:delete-outlined",
              onClick: e(ne)
            }, {
              default: i(() => [
                _(" 批量删除")
              ]),
              _: 1
            }, 8, ["onClick"])), [
              [je, e(ye).length > 0]
            ]) : d("", !0),
            e(C).super_query ? (n(), a(po, {
              key: 5,
              ref_key: "superQueryButtonRef",
              ref: M,
              online: "",
              status: e(U),
              onSearch: e(W)
            }, null, 8, ["status", "onSearch"])) : d("", !0)
          ]),
          expandedRowRender: i(() => [
            p(B, {
              activeKey: c.tabIndex,
              "onUpdate:activeKey": r[0] || (r[0] = (o) => c.tabIndex = o)
            }, {
              default: i(() => [
                l.value.length ? (n(!0), y(D, { key: 0 }, K(c.tabNav, (o, f) => (n(), a(T, {
                  tab: o.tableTxt,
                  key: f + ""
                }, {
                  default: i(() => [
                    c.tabIndex == f ? (n(), a(go, {
                      key: 0,
                      subTableId: o.id,
                      mTableSelectedRcordId: l.value[0],
                      subTableName: o.tableName
                    }, null, 8, ["subTableId", "mTableSelectedRcordId", "subTableName"])) : d("", !0)
                  ]),
                  _: 2
                }, 1032, ["tab"]))), 128)) : d("", !0)
              ]),
              _: 1
            }, 8, ["activeKey"])
          ]),
          fileSlot: i(({ text: o }) => [
            o ? (n(), a(m, {
              key: 1,
              ghost: !0,
              type: "primary",
              preIcon: "ant-design:download",
              size: "small",
              onClick: (f) => e(ke)(o)
            }, {
              default: i(() => [
                _(" 下载 ")
              ]),
              _: 2
            }, 1032, ["onClick"])) : (n(), y("span", yo, "无文件"))
          ]),
          imgSlot: i(({ text: o }) => [
            o ? (n(), y("img", {
              key: 1,
              src: e(Ie)(o),
              alt: "图片不存在",
              class: "online-cell-image",
              onClick: (f) => e(ve)(o)
            }, null, 8, ho)) : (n(), y("span", _o, "无图片"))
          ]),
          htmlSlot: i(({ text: o }) => [
            N("div", { innerHTML: o }, null, 8, Co)
          ]),
          pcaSlot: i(({ text: o }) => [
            N("div", {
              title: e(A)(o)
            }, R(e(A)(o)), 9, ko)
          ]),
          dateSlot: i(({ text: o }) => [
            N("span", null, R(e(Se)(o)), 1)
          ]),
          action: i(({ record: o }) => [
            p(e(Ze), {
              actions: e(ue)(o),
              dropDownActions: e(ce)(o)
            }, null, 8, ["actions", "dropDownActions"])
          ]),
          _: 1
        }, 8, ["loading", "columns", "dataSource", "pagination", "rowSelection", "actionColumn", "scroll", "onTableRedo", "class", "onChange", "expandedRowKeys"])),
        p(oo, {
          id: e(b),
          onRegister: e(oe),
          onSuccess: Ae,
          onFormConfig: Pe
        }, null, 8, ["id", "onRegister"]),
        p(J, {
          id: e(b),
          onRegister: e(me)
        }, null, 8, ["id", "onRegister"]),
        p(no, {
          onRegister: e(ie),
          url: e(ee)(),
          onOk: e(u),
          online: ""
        }, null, 8, ["onRegister", "url", "onOk"]),
        p(k, Ve(e(S).model, qe(e(S).on)), {
          default: i(() => [
            (n(), a(Ue(e(S).is), We(Ge(e(S).params)), null, 16))
          ]),
          _: 1
        }, 16),
        p(to, {
          onRegister: e(X),
          onSuccess: e(u)
        }, null, 8, ["onRegister", "onSuccess"]),
        p(J, {
          id: e(we),
          onRegister: e(Re),
          defaultFullscreen: !1
        }, null, 8, ["id", "onRegister"]),
        p(so, {
          ref_key: "onlinePopModalRef",
          ref: Oe,
          id: e(Q),
          onRegister: e(xe),
          onSuccess: e(u),
          request: "",
          topTip: ""
        }, null, 8, ["id", "onRegister", "onSuccess"])
      ]);
    };
  }
};
export {
  Ft as default
};
