// @jeecg/online@3.6.2-beta/OnlCgformInnerSubTable-fbca73e0.mjs
var M = (h, c, a) => new Promise((d, n) => {
  var u = (i) => {
    try {
      g(a.next(i));
    } catch (f) {
      n(f);
    }
  }, s = (i) => {
    try {
      g(a.throw(i));
    } catch (f) {
      n(f);
    }
  }, g = (i) => i.done ? d(i.value) : Promise.resolve(i.value).then(u, s);
  g((a = a.apply(h, c)).next());
});
import { watch as ne, resolveComponent as w, openBlock as l, createElementBlock as _, Fragment as ie, unref as e, createBlock as C, normalizeClass as le, withCtx as r, createTextVNode as re, createElementVNode as T, toDisplayString as y, createCommentVNode as ae, createVNode as p, mergeProps as se, toHandlers as me, resolveDynamicComponent as pe, normalizeProps as ce, guardReactiveProps as de } from "vue";
import { BasicTable as ue } from "/@/components/Table";
import { useMessage as ge } from "/@/hooks/web/useMessage";
import fe from "./OnlineCustomModal-518ad33c.mjs";
import k from "./OnlineDetailModal-9675390d.mjs";
import be from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as _e, a as he } from "./useListButton-cdbab702.mjs";
import { u as Ce, a as Te, O as Se } from "./useExtendComponent-325fe3f1.mjs";
import { u as Re } from "./useOnlinePopEvent-687070b7.mjs";
import { I as Ie } from "./constant-754f1a9d.mjs";
import "/@/components/Table/src/const";
import "/@/components/Modal";
import "/@/utils/http/axios";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "lodash-es";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
import "./OnlineFormDetail-0e7294ca.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
const Me = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, we = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, ye = ["src", "onClick"], ke = ["innerHTML"], ve = ["title"], Vo = {
  __name: "OnlCgformInnerSubTable",
  props: ["subTableId", "subTableName", "mTableSelectedRcordId"],
  setup(h) {
    const c = h, { createMessage: a } = ge(), {
      ID: d,
      onlineTableContext: n,
      loading: u,
      reload: s,
      dataSource: g,
      handleSpecialConfig: i,
      getColumnList: f,
      handleChangeInTable: v,
      loadData: O,
      onlineExtConfigJson: S,
      registerCustomModal: B,
      tableReloading: P
    } = _e({ code: c.subTableId });
    n.isInnerSubTable = !0, n.innerSubTableName = c.subTableName, n.mTableSelectedRcordId = c.mTableSelectedRcordId, d.value || a.warning("地址错误, 配置ID不存在!");
    let { initCgEnhanceJs: E } = Ce(n);
    const {
      importUrl: x,
      registerImportModal: L,
      initButtonList: N,
      initButtonSwitch: D,
      registerDetailModal: z,
      registerBpmModal: Oe
    } = he(n, S), {
      columns: J,
      enableScrollBar: V,
      tableScroll: F,
      downloadRowFile: H,
      getImgView: A,
      getPcaText: R,
      getFormatDate: $,
      handleColumnResult: j,
      hrefComponent: b,
      viewOnlineCellImage: U,
      hrefMainTableId: q,
      registerOnlineHrefModal: K,
      registerPopModal: G,
      openPopModal: Q,
      onlinePopModalRef: W,
      popTableId: I
    } = Te(n, S);
    ne(
      d,
      () => {
        X();
      },
      { immediate: !0 }
    );
    function X() {
      return M(this, null, function* () {
        u.value = !0;
        let o = yield f(Ie);
        Y(o), yield O(), u.value = !1, n.execButtonEnhance("setup");
      });
    }
    function Y(o) {
      let m = E(o.enhanceJs);
      n.EnhanceJS = m, N(o.cgButtonList), D(o.hideColumns), j(o), i(o);
    }
    function Z(o) {
      I.value = o.id;
      let m = {
        title: o.describe
      };
      o.record && o.record.id && (m.record = o.record, m.isUpdate = !0), Q(!0, m);
    }
    return Re(Z), (o, m) => {
      const ee = w("a-button"), oe = w("a-modal");
      return l(), _(ie, null, [
        e(P) ? ae("", !0) : (l(), C(e(ue), {
          key: 0,
          ref: "onlineTable",
          rowKey: "jeecg_row_key",
          canResize: !0,
          bordered: !0,
          showIndexColumn: !1,
          loading: e(u),
          columns: e(J),
          dataSource: e(g),
          pagination: !1,
          showActionColumn: !1,
          showTableSetting: !1,
          clickToRowSelect: !1,
          scroll: e(F),
          onTableRedo: e(s),
          class: le({ "j-table-force-nowrap": e(V) }),
          onChange: e(v)
        }, {
          fileSlot: r(({ text: t }) => [
            t ? (l(), C(ee, {
              key: 1,
              ghost: !0,
              type: "primary",
              preIcon: "ant-design:download",
              size: "small",
              onClick: (te) => e(H)(t)
            }, {
              default: r(() => [
                re(" 下载 ")
              ]),
              _: 2
            }, 1032, ["onClick"])) : (l(), _("span", Me, "无文件"))
          ]),
          imgSlot: r(({ text: t }) => [
            t ? (l(), _("img", {
              key: 1,
              src: e(A)(t),
              alt: "图片不存在",
              class: "online-cell-image",
              onClick: (te) => e(U)(t)
            }, null, 8, ye)) : (l(), _("span", we, "无图片"))
          ]),
          htmlSlot: r(({ text: t }) => [
            T("div", { innerHTML: t }, null, 8, ke)
          ]),
          pcaSlot: r(({ text: t }) => [
            T("div", {
              title: e(R)(t)
            }, y(e(R)(t)), 9, ve)
          ]),
          dateSlot: r(({ text: t }) => [
            T("span", null, y(e($)(t)), 1)
          ]),
          _: 1
        }, 8, ["loading", "columns", "dataSource", "scroll", "onTableRedo", "class", "onChange"])),
        p(k, {
          id: e(d),
          onRegister: e(z)
        }, null, 8, ["id", "onRegister"]),
        p(be, {
          onRegister: e(L),
          url: e(x)(),
          onOk: e(s),
          online: ""
        }, null, 8, ["onRegister", "url", "onOk"]),
        p(oe, se(e(b).model, me(e(b).on)), {
          default: r(() => [
            (l(), C(pe(e(b).is), ce(de(e(b).params)), null, 16))
          ]),
          _: 1
        }, 16),
        p(fe, {
          onRegister: e(B),
          onSuccess: e(s)
        }, null, 8, ["onRegister", "onSuccess"]),
        p(k, {
          id: e(q),
          onRegister: e(K),
          defaultFullscreen: !1
        }, null, 8, ["id", "onRegister"]),
        p(Se, {
          ref_key: "onlinePopModalRef",
          ref: W,
          id: e(I),
          onRegister: e(G),
          onSuccess: e(s),
          request: "",
          topTip: ""
        }, null, 8, ["id", "onRegister", "onSuccess"])
      ], 64);
    };
  }
};
export {
  Vo as default
};
