// @jeecg/online@3.6.2-beta/OnlCgformTabList-a9e1e36a.mjs
var w = (B, R, r) => new Promise((c, k) => {
  var y = (a) => {
    try {
      C(r.next(a));
    } catch (_) {
      k(_);
    }
  }, p = (a) => {
    try {
      C(r.throw(a));
    } catch (_) {
      k(_);
    }
  }, C = (a) => a.done ? c(a.value) : Promise.resolve(a.value).then(y, p);
  C((r = r.apply(B, R)).next());
});
import { defineComponent as Be, ref as xe, watch as De, resolveComponent as L, openBlock as n, createElementBlock as f, createVNode as s, unref as e, createBlock as l, normalizeClass as Ee, withCtx as i, createTextVNode as d, createCommentVNode as u, Fragment as P, renderList as Le, toDisplayString as T, withDirectives as Pe, vShow as Fe, createElementVNode as O, mergeProps as Ae, toHandlers as Qe, resolveDynamicComponent as Je, normalizeProps as ze, guardReactiveProps as He } from "vue";
import { BasicTable as Ne, TableAction as Ve } from "/@/components/Table";
import { useMessage as $e } from "/@/hooks/web/useMessage";
import qe from "./OnlineCustomModal-518ad33c.mjs";
import je from "./OnlineTabAutoModal-6f636ee2.mjs";
import F from "./OnlineTabDetailModal-7356d3a8.mjs";
import Ke from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as Ue, a as We } from "./useListButton-cdbab702.mjs";
import { u as Ge, a as Xe, O as Ye, g as Ze } from "./useExtendComponent-325fe3f1.mjs";
import eo from "./OnlineQueryForm-1bfc8320.mjs";
import oo from "./SuperQuery-5d1ca957.mjs";
import { u as to } from "./useOnlinePopEvent-687070b7.mjs";
import { T as no } from "./constant-754f1a9d.mjs";
import "/@/components/Modal";
import "/@/utils/http/axios";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "lodash-es";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
import "./OnlineTabForm-512d3951.mjs";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "./OnlineTabFormDetail-5a70219d.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
import "./SuperQueryValComponent.vue_vue_type_script_lang-e9196096.mjs";
import "/@/utils/is";
const io = { class: "p-2" }, ro = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, lo = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, ao = ["src", "onClick"], so = ["innerHTML"], co = ["title"], Rt = /* @__PURE__ */ Be({
  __name: "OnlCgformTabList",
  setup(B) {
    const { createMessage: R } = $e(), {
      ID: r,
      onlineTableContext: c,
      onlineQueryFormOuter: k,
      loading: y,
      reload: p,
      dataSource: C,
      pagination: a,
      handleSpecialConfig: _,
      getColumnList: A,
      handleChangeInTable: Q,
      loadData: J,
      superQueryButtonRef: v,
      superQueryStatus: z,
      handleSuperQuery: H,
      onlineExtConfigJson: x,
      handleFormConfig: N,
      registerCustomModal: V,
      tableReloading: $
    } = Ue();
    if (!r.value)
      throw R.warning("地址错误, 配置ID不存在!"), new Error("地址错误, 配置ID不存在!");
    let { initCgEnhanceJs: q } = Ge(c);
    const {
      buttonSwitch: h,
      cgTopButtonList: I,
      importUrl: j,
      registerModal: K,
      handleAdd: U,
      handleBatchDelete: W,
      registerImportModal: G,
      onImportExcel: X,
      onExportExcel: Y,
      cgButtonJsHandler: Z,
      cgButtonActionHandler: ee,
      getDropDownActions: oe,
      getActions: te,
      initButtonList: ne,
      initButtonSwitch: ie,
      registerDetailModal: re,
      registerBpmModal: po
    } = We(c, x), M = xe(!1);
    function le() {
      return w(this, null, function* () {
        try {
          M.value = !0, yield Y();
        } finally {
          setTimeout(() => M.value = !1, 1500);
        }
      });
    }
    const {
      columns: ae,
      actionColumn: se,
      selectedKeys: ce,
      rowSelection: pe,
      enableScrollBar: ue,
      tableScroll: me,
      downloadRowFile: de,
      getImgView: ge,
      getPcaText: D,
      getFormatDate: fe,
      handleColumnResult: ye,
      hrefComponent: b,
      viewOnlineCellImage: Ce,
      hrefMainTableId: _e,
      registerOnlineHrefModal: he,
      registerPopModal: ke,
      openPopModal: be,
      onlinePopModalRef: Se,
      popTableId: E
    } = Xe(c, x);
    De(
      r,
      () => {
        we();
      },
      { immediate: !0 }
    );
    function we() {
      return w(this, null, function* () {
        y.value = !0;
        let t = yield A(no);
        Te(t), yield J(), y.value = !1, c.execButtonEnhance("setup");
      });
    }
    function Te(t) {
      let g = q(t.enhanceJs);
      c.EnhanceJS = g, ne(t.cgButtonList), ie(t.hideColumns), ye(t), _(t);
    }
    function Re(t) {
      c.queryParam = t, p({ mode: "search" });
    }
    function ve(t) {
      return w(this, null, function* () {
        yield Ze(v), v.value.init(t);
      });
    }
    function Ie(t) {
      E.value = t.id;
      let g = {
        title: t.describe
      };
      t.record && t.record.id && (g.record = t.record, g.isUpdate = !0), be(!0, g);
    }
    return to(Ie), (t, g) => {
      const m = L("a-button"), Me = L("a-modal");
      return n(), f("div", io, [
        s(eo, {
          ref_key: "onlineQueryFormOuter",
          ref: k,
          id: e(r),
          onSearch: Re,
          onLoaded: ve
        }, null, 8, ["id"]),
        e($) ? u("", !0) : (n(), l(e(Ne), {
          key: 0,
          ref: "onlineTable",
          rowKey: "jeecg_row_key",
          canResize: !0,
          bordered: !0,
          showIndexColumn: !1,
          loading: e(y),
          columns: e(ae),
          dataSource: e(C),
          pagination: e(a),
          rowSelection: e(pe),
          actionColumn: e(se),
          showTableSetting: !0,
          clickToRowSelect: !1,
          scroll: e(me),
          onTableRedo: e(p),
          class: Ee({ "j-table-force-nowrap": e(ue) }),
          onChange: e(Q)
        }, {
          tableTitle: i(() => [
            e(h).add ? (n(), l(m, {
              key: 0,
              preIcon: "ant-design:plus-outlined",
              type: "primary",
              onClick: e(U)
            }, {
              default: i(() => [
                d("新增")
              ]),
              _: 1
            }, 8, ["onClick"])) : u("", !0),
            e(h).import ? (n(), l(m, {
              key: 1,
              preIcon: "ant-design:import-outlined",
              type: "primary",
              onClick: e(X)
            }, {
              default: i(() => [
                d("导入")
              ]),
              _: 1
            }, 8, ["onClick"])) : u("", !0),
            e(h).export ? (n(), l(m, {
              key: 2,
              preIcon: "ant-design:export-outlined",
              type: "primary",
              loading: M.value,
              onClick: le
            }, {
              default: i(() => [
                d(" 导出")
              ]),
              _: 1
            }, 8, ["loading"])) : u("", !0),
            e(I) && e(I).length > 0 ? (n(!0), f(P, { key: 3 }, Le(e(I), (o, S) => (n(), f(P, null, [
              o.optType == "js" ? (n(), l(m, {
                key: "cgbtn" + S,
                onClick: (Oe) => e(Z)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: i(() => [
                  d(T(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : o.optType == "action" ? (n(), l(m, {
                key: "cgbtn" + S,
                onClick: (Oe) => e(ee)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: i(() => [
                  d(T(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : u("", !0)
            ], 64))), 256)) : u("", !0),
            e(h).batch_delete ? Pe((n(), l(m, {
              key: 4,
              preIcon: "ant-design:delete-outlined",
              onClick: e(W)
            }, {
              default: i(() => [
                d(" 批量删除")
              ]),
              _: 1
            }, 8, ["onClick"])), [
              [Fe, e(ce).length > 0]
            ]) : u("", !0),
            e(h).super_query ? (n(), l(oo, {
              key: 5,
              ref_key: "superQueryButtonRef",
              ref: v,
              online: "",
              status: e(z),
              onSearch: e(H)
            }, null, 8, ["status", "onSearch"])) : u("", !0)
          ]),
          fileSlot: i(({ text: o }) => [
            o ? (n(), l(m, {
              key: 1,
              ghost: !0,
              type: "primary",
              preIcon: "ant-design:download",
              size: "small",
              onClick: (S) => e(de)(o)
            }, {
              default: i(() => [
                d(" 下载 ")
              ]),
              _: 2
            }, 1032, ["onClick"])) : (n(), f("span", ro, "无文件"))
          ]),
          imgSlot: i(({ text: o }) => [
            o ? (n(), f("img", {
              key: 1,
              src: e(ge)(o),
              alt: "图片不存在",
              class: "online-cell-image",
              onClick: (S) => e(Ce)(o)
            }, null, 8, ao)) : (n(), f("span", lo, "无图片"))
          ]),
          htmlSlot: i(({ text: o }) => [
            O("div", { innerHTML: o }, null, 8, so)
          ]),
          pcaSlot: i(({ text: o }) => [
            O("div", {
              title: e(D)(o)
            }, T(e(D)(o)), 9, co)
          ]),
          dateSlot: i(({ text: o }) => [
            O("span", null, T(e(fe)(o)), 1)
          ]),
          action: i(({ record: o }) => [
            s(e(Ve), {
              actions: e(te)(o),
              dropDownActions: e(oe)(o)
            }, null, 8, ["actions", "dropDownActions"])
          ]),
          _: 1
        }, 8, ["loading", "columns", "dataSource", "pagination", "rowSelection", "actionColumn", "scroll", "onTableRedo", "class", "onChange"])),
        s(je, {
          id: e(r),
          onRegister: e(K),
          onSuccess: e(p),
          onFormConfig: e(N)
        }, null, 8, ["id", "onRegister", "onSuccess", "onFormConfig"]),
        s(F, {
          id: e(r),
          onRegister: e(re)
        }, null, 8, ["id", "onRegister"]),
        s(Ke, {
          onRegister: e(G),
          url: e(j)(),
          onOk: e(p),
          online: ""
        }, null, 8, ["onRegister", "url", "onOk"]),
        s(Me, Ae(e(b).model, Qe(e(b).on)), {
          default: i(() => [
            (n(), l(Je(e(b).is), ze(He(e(b).params)), null, 16))
          ]),
          _: 1
        }, 16),
        s(qe, {
          onRegister: e(V),
          onSuccess: e(p)
        }, null, 8, ["onRegister", "onSuccess"]),
        s(F, {
          id: e(_e),
          onRegister: e(he),
          defaultFullscreen: !1
        }, null, 8, ["id", "onRegister"]),
        s(Ye, {
          ref_key: "onlinePopModalRef",
          ref: Se,
          id: e(E),
          onRegister: e(ke),
          onSuccess: e(p),
          request: "",
          topTip: ""
        }, null, 8, ["id", "onRegister", "onSuccess"])
      ]);
    };
  }
});
export {
  Rt as default
};
