// @jeecg/online@3.6.2-beta/OnlCgformErpSubTable-54b50eec.mjs
var O = (R, w, c) => new Promise((I, u) => {
  var h = (r) => {
    try {
      C(c.next(r));
    } catch (s) {
      u(s);
    }
  }, n = (r) => {
    try {
      C(c.throw(r));
    } catch (s) {
      u(s);
    }
  }, C = (r) => r.done ? I(r.value) : Promise.resolve(r.value).then(h, n);
  C((c = c.apply(R, w)).next());
});
import { ref as B, provide as Fe, watch as K, resolveComponent as J, openBlock as i, createElementBlock as y, Fragment as x, createVNode as p, unref as e, createBlock as d, normalizeClass as De, withCtx as l, createTextVNode as g, createCommentVNode as f, renderList as Le, toDisplayString as T, withDirectives as Pe, vShow as Ae, createElementVNode as F, mergeProps as He, toHandlers as Ke, resolveDynamicComponent as Je, normalizeProps as Ne, guardReactiveProps as ze } from "vue";
import { BasicTable as Ve, TableAction as $e } from "/@/components/Table";
import "/@/hooks/web/useMessage";
import je from "./OnlineAutoModal-510991eb.mjs";
import qe from "./OnlineCustomModal-518ad33c.mjs";
import N from "./OnlineDetailModal-9675390d.mjs";
import Qe from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as Ue, a as We } from "./useListButton-cdbab702.mjs";
import { u as Ge, a as Xe, O as Ye } from "./useExtendComponent-325fe3f1.mjs";
import Ze from "./OnlineQueryForm-1bfc8320.mjs";
import { u as eo } from "./useOnlinePopEvent-687070b7.mjs";
import { E as oo } from "./constant-754f1a9d.mjs";
import { cloneDeep as to } from "lodash-es";
import "/@/components/Modal";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "/@/utils/http/axios";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "./index-0e3ab3af.mjs";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
import "vue-router";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "./OnlineFormDetail-0e7294ca.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
const no = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, io = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, lo = ["src", "onClick"], ro = ["innerHTML"], ao = ["title"], Tt = {
  __name: "OnlCgformErpSubTable",
  props: ["data", "mainTableSelectedRowRcord"],
  emits: ["getSource"],
  setup(R, { emit: w }) {
    const c = R, I = w, u = B(!1), {
      ID: h,
      onlineTableContext: n,
      onlineQueryFormOuter: C,
      loading: r,
      reload: s,
      dataSource: b,
      pagination: z,
      handleSpecialConfig: V,
      getColumnList: co,
      handleChangeInTable: $,
      loadData: j,
      onlineExtConfigJson: D,
      handleFormConfig: q,
      registerCustomModal: Q,
      tableReloading: U
    } = Ue({ code: c.data.code, themeTemplate: oo });
    n.isErpSubTable = !0;
    let { initCgEnhanceJs: W } = Ge(n);
    const {
      buttonSwitch: k,
      cgLinkButtonList: so,
      cgTopButtonList: M,
      importUrl: G,
      registerModal: X,
      handleAdd: Y,
      handleEdit: uo,
      handleBatchDelete: Z,
      registerImportModal: ee,
      onImportExcel: oe,
      onExportExcel: te,
      cgButtonJsHandler: ne,
      cgButtonActionHandler: ie,
      cgButtonLinkHandler: po,
      handleSubmitFlow: mo,
      getDropDownActions: le,
      getActions: re,
      initButtonList: ae,
      initButtonSwitch: ce,
      registerDetailModal: se,
      registerBpmModal: go
    } = We(n, D), E = B(!1);
    function ue() {
      return O(this, null, function* () {
        try {
          E.value = !0, yield te();
        } finally {
          setTimeout(() => E.value = !1, 1500);
        }
      });
    }
    const {
      columns: pe,
      actionColumn: de,
      selectedKeys: me,
      rowSelection: ge,
      enableScrollBar: fe,
      tableScroll: ye,
      downloadRowFile: he,
      getImgView: Ce,
      getPcaText: L,
      getFormatDate: be,
      handleColumnResult: ke,
      hrefComponent: S,
      viewOnlineCellImage: Se,
      hrefMainTableId: _e,
      registerOnlineHrefModal: ve,
      registerPopModal: Te,
      openPopModal: Re,
      onlinePopModalRef: we,
      popTableId: P
    } = Xe(n, D);
    Ie(to(c.data));
    const A = B(null), _ = c.data.foreignKeys;
    let H;
    if (_ != null && _.length) {
      const t = _[0], a = t.field;
      H = t.key, n.foreignKeyField = a;
    } else
      n.foreignKeyField = null, n.foreignKeyValue = null;
    Fe("foreignkey", A), K(
      () => c.mainTableSelectedRowRcord,
      (t) => {
        if (t) {
          if (n.foreignKeyField) {
            const a = t[H];
            n.foreignKeyValue = a, A.value = { field: n.foreignKeyField, value: a };
          }
          u.value = !0, r.value = !0, j().finally(() => {
            r.value = !1;
          });
        } else
          u.value = !1, b.value = [];
      },
      {
        immediate: !0
      }
    ), K(
      () => b.value,
      () => {
        I("getSource", c.data.currentTableName, b.value);
      },
      { immediate: !0 }
    );
    function Ie(t) {
      return O(this, null, function* () {
        Me(t), r.value = !1, n.execButtonEnhance("setup");
      });
    }
    function Me(t) {
      let a = W(t.enhanceJs);
      n.EnhanceJS = a, ae(t.cgButtonList), ce(t.hideColumns), ke(t), V(t);
    }
    function Ee(t) {
      n.queryParam = t, s({ mode: "search" });
    }
    function Oe(t) {
      P.value = t.id;
      let a = {
        title: t.describe
      };
      t.record && t.record.id && (a.record = t.record, a.isUpdate = !0), Re(!0, a);
    }
    return eo(Oe), (t, a) => {
      const m = J("a-button"), Be = J("a-modal");
      return i(), y(x, null, [
        p(Ze, {
          ref_key: "onlineQueryFormOuter",
          ref: C,
          id: e(h),
          onSearch: Ee
        }, null, 8, ["id"]),
        e(U) ? f("", !0) : (i(), d(e(Ve), {
          key: 0,
          ref: "onlineTable",
          rowKey: "jeecg_row_key",
          canResize: !0,
          bordered: !0,
          showIndexColumn: !1,
          loading: e(r),
          columns: e(pe),
          dataSource: e(b),
          pagination: e(z),
          rowSelection: e(ge),
          actionColumn: e(de),
          showTableSetting: !0,
          clickToRowSelect: !1,
          scroll: e(ye),
          onTableRedo: e(s),
          class: De({ "j-table-force-nowrap": e(fe) }),
          onChange: e($),
          minHeight: 300
        }, {
          tableTitle: l(() => [
            u.value && e(k).add ? (i(), d(m, {
              key: 0,
              preIcon: "ant-design:plus-outlined",
              type: "primary",
              onClick: e(Y)
            }, {
              default: l(() => [
                g("新增")
              ]),
              _: 1
            }, 8, ["onClick"])) : f("", !0),
            u.value && e(k).import ? (i(), d(m, {
              key: 1,
              preIcon: "ant-design:import-outlined",
              type: "primary",
              onClick: e(oe)
            }, {
              default: l(() => [
                g("导入")
              ]),
              _: 1
            }, 8, ["onClick"])) : f("", !0),
            u.value && e(k).export ? (i(), d(m, {
              key: 2,
              preIcon: "ant-design:export-outlined",
              type: "primary",
              loading: E.value,
              onClick: ue
            }, {
              default: l(() => [
                g(" 导出")
              ]),
              _: 1
            }, 8, ["loading"])) : f("", !0),
            u.value && e(M) && e(M).length > 0 ? (i(!0), y(x, { key: 3 }, Le(e(M), (o, v) => (i(), y(x, null, [
              o.optType == "js" ? (i(), d(m, {
                key: "cgbtn" + v,
                onClick: (xe) => e(ne)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: l(() => [
                  g(T(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : o.optType == "action" ? (i(), d(m, {
                key: "cgbtn" + v,
                onClick: (xe) => e(ie)(o.buttonCode),
                type: "primary",
                preIcon: o.buttonIcon ? "ant-design:" + o.buttonIcon : ""
              }, {
                default: l(() => [
                  g(T(o.buttonName), 1)
                ]),
                _: 2
              }, 1032, ["onClick", "preIcon"])) : f("", !0)
            ], 64))), 256)) : f("", !0),
            e(k).batch_delete ? Pe((i(), d(m, {
              key: 4,
              preIcon: "ant-design:delete-outlined",
              onClick: e(Z)
            }, {
              default: l(() => [
                g(" 批量删除")
              ]),
              _: 1
            }, 8, ["onClick"])), [
              [Ae, e(me).length > 0]
            ]) : f("", !0)
          ]),
          fileSlot: l(({ text: o }) => [
            o ? (i(), d(m, {
              key: 1,
              ghost: !0,
              type: "primary",
              preIcon: "ant-design:download",
              size: "small",
              onClick: (v) => e(he)(o)
            }, {
              default: l(() => [
                g(" 下载 ")
              ]),
              _: 2
            }, 1032, ["onClick"])) : (i(), y("span", no, "无文件"))
          ]),
          imgSlot: l(({ text: o }) => [
            o ? (i(), y("img", {
              key: 1,
              src: e(Ce)(o),
              alt: "图片不存在",
              class: "online-cell-image",
              onClick: (v) => e(Se)(o)
            }, null, 8, lo)) : (i(), y("span", io, "无图片"))
          ]),
          htmlSlot: l(({ text: o }) => [
            F("div", { innerHTML: o }, null, 8, ro)
          ]),
          pcaSlot: l(({ text: o }) => [
            F("div", {
              title: e(L)(o)
            }, T(e(L)(o)), 9, ao)
          ]),
          dateSlot: l(({ text: o }) => [
            F("span", null, T(e(be)(o)), 1)
          ]),
          action: l(({ record: o }) => [
            p(e($e), {
              actions: e(re)(o),
              dropDownActions: e(le)(o)
            }, null, 8, ["actions", "dropDownActions"])
          ]),
          _: 1
        }, 8, ["loading", "columns", "dataSource", "pagination", "rowSelection", "actionColumn", "scroll", "onTableRedo", "class", "onChange"])),
        p(je, {
          id: e(h),
          onRegister: e(X),
          onSuccess: e(s),
          onFormConfig: e(q)
        }, null, 8, ["id", "onRegister", "onSuccess", "onFormConfig"]),
        p(N, {
          id: e(h),
          onRegister: e(se)
        }, null, 8, ["id", "onRegister"]),
        p(Qe, {
          onRegister: e(ee),
          url: e(G)(),
          onOk: e(s),
          online: ""
        }, null, 8, ["onRegister", "url", "onOk"]),
        p(Be, He(e(S).model, Ke(e(S).on)), {
          default: l(() => [
            (i(), d(Je(e(S).is), Ne(ze(e(S).params)), null, 16))
          ]),
          _: 1
        }, 16),
        p(qe, {
          onRegister: e(Q),
          onSuccess: e(s)
        }, null, 8, ["onRegister", "onSuccess"]),
        p(N, {
          id: e(_e),
          onRegister: e(ve),
          defaultFullscreen: !1
        }, null, 8, ["id", "onRegister"]),
        p(Ye, {
          ref_key: "onlinePopModalRef",
          ref: we,
          id: e(P),
          onRegister: e(Te),
          onSuccess: e(s),
          request: "",
          topTip: ""
        }, null, 8, ["id", "onRegister", "onSuccess"])
      ], 64);
    };
  }
};
export {
  Tt as default
};
