// @jeecg/online@3.6.2-beta/OnlineQueryForm-1bfc8320.mjs
var ce = Object.defineProperty;
var J = Object.getOwnPropertySymbols;
var me = Object.prototype.hasOwnProperty, ue = Object.prototype.propertyIsEnumerable;
var q = (i, s, l) => s in i ? ce(i, s, { enumerable: !0, configurable: !0, writable: !0, value: l }) : i[s] = l, E = (i, s) => {
  for (var l in s || (s = {}))
    me.call(s, l) && q(i, l, s[l]);
  if (J)
    for (var l of J(s))
      ue.call(s, l) && q(i, l, s[l]);
  return i;
};
var D = (i, s, l) => new Promise((c, w) => {
  var V = (f) => {
    try {
      _(l.next(f));
    } catch (S) {
      w(S);
    }
  }, F = (f) => {
    try {
      _(l.throw(f));
    } catch (S) {
      w(S);
    }
  }, _ = (f) => f.done ? c(f.value) : Promise.resolve(f.value).then(V, F);
  _((l = l.apply(i, s)).next());
});
import { BasicForm as pe, useForm as de } from "/@/components/Form/index";
import { ref as O, reactive as fe, watch as Q, toRaw as R, resolveComponent as I, openBlock as W, createElementBlock as G, createVNode as p, withCtx as b, createElementVNode as U, createTextVNode as T, toDisplayString as he, createCommentVNode as K, pushScopeId as ge, popScopeId as _e } from "vue";
import { defHttp as ye } from "/@/utils/http/axios";
import { useMessage as ve } from "/@/hooks/web/useMessage";
import { j as be, g as we, k as z, m as Fe, F as B, L as Se, n as xe, o as X } from "./useExtendComponent-325fe3f1.mjs";
import { L as ke } from "./constant-754f1a9d.mjs";
import { _ as Oe } from "./index-0e3ab3af.mjs";
import "/@/components/Form/src/componentMap";
import "/@/utils/propTypes";
import "@ant-design/icons-vue";
import "/@/components/Modal";
import "lodash-es";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/common/compUtils";
import "/@/components/Table";
import "/@/hooks/system/useListPage";
import "vue-router";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "/@/components/Loading";
import "/@/utils/auth";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "ant-design-vue";
import "@vueuse/core";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const Le = {
  name: "OnlineQueryForm",
  components: {
    BasicForm: pe
  },
  props: {
    id: {
      type: String,
      default: ""
    }
  },
  emits: ["search", "loaded"],
  setup(i, { emit: s }) {
    const l = "/online/cgform/api/getQueryInfoVue3/", c = O(null), w = O([]), V = O({ xs: 24, sm: 24, md: 12, lg: 6, xl: 6 }), F = O(!1), _ = O(!1), f = O({}), S = O([]), { createMessage: Y } = ve(), { linkTableCard2Select: N } = be(), j = O(80), o = fe({
      config: {},
      cache: {},
      param: {},
      status: !1
    });
    Q(
      () => o.status,
      (t) => D(this, null, function* () {
        const { config: e, cache: n, param: y } = R(o);
        let u = Object.assign({}, e, n, y);
        yield $(u);
      }),
      { immediate: !0, deep: !0 }
    );
    function m(t, e) {
      return D(this, null, function* () {
        o.cache = E({}, t), o.param = E({}, e), o.status = !o.status;
      });
    }
    Q(
      () => i.id,
      (t) => {
        t ? Z() : w.value = [];
      },
      { immediate: !0 }
    );
    function d(t) {
      return D(this, null, function* () {
        let e = [], n = {}, y = Object.keys(t), u = -1;
        for (let a of y) {
          const r = t[a];
          a === "sys_org_code" && (r.fieldExtendJson || (r.fieldExtendJson = '{"store":"orgCode"}'));
          let h = r.view;
          if (z[h] && (r.view = z[h]), yield Fe(a, r, n), r.mode == "group" && (h == "date" || h == "datetime" || h == "number")) {
            let L = B.createSlotFormSchema(a, r);
            e.push(L);
          } else if (r.view === Se) {
            let L = xe(r, a);
            for (let k of L) {
              let C = B.createFormSchema(k.key, k), H = X(e, k.key);
              H == -1 ? e.push(C) : e[H] = C;
            }
          } else if (X(e, a) == -1) {
            let k = B.createFormSchema(a, r);
            e.push(k);
          }
          let x = r.fieldExtendJson;
          x && (x = JSON.parse(x), x.labelLength && (u > -1 ? u = x.labelLength > u ? x.labelLength : u : u = x.labelLength));
        }
        u == -1 && (u = ke), e.sort(function(a, r) {
          return a.order - r.order;
        });
        let g = [];
        e.length > 2 && (F.value = !0);
        let v = [];
        for (let a = 0; a < e.length; a++) {
          let r = e[a];
          r.setFormRef(c), r.noChange(), r.asSearchForm(), a > 1 && (v.push(r.field), r.isHidden());
          let h = r.getFormItemSchema();
          r.slot == "groupDatetime" && (h.colProps = { xs: 24, sm: 24, md: 12, lg: 8, xl: 8 }), N(h), g.push(h);
        }
        S.value = v, w.value = g, o.config = E({}, n), o.status = !o.status, setTimeout(() => {
          const a = u * 14 + u + 24;
          j.value = a;
        }, 0);
      });
    }
    function Z() {
      return D(this, null, function* () {
        let t = yield ae(), e = te(t);
        s("loaded", t);
        let { formProperties: n, hasField: y } = ee(e, t);
        if (y == !1) {
          w.value = [];
          return;
        }
        yield d(n);
      });
    }
    function $(t) {
      return D(this, null, function* () {
        yield we(c), yield A(t), Object.keys(t).length > 0 && M();
      });
    }
    function ee(t, e) {
      const { searchFieldList: n, joinQuery: y, table: u } = e;
      let g = !1, v = {};
      return t && Object.keys(t).map((a) => {
        n.indexOf(a) >= 0 && (y == !0 ? a.indexOf("@") < 0 ? (v[u + "@" + a] = t[a], g = !0) : (v[a] = t[a], g = !0) : a.indexOf("@") < 0 && (v[a] = t[a], g = !0));
      }), {
        formProperties: v,
        hasField: g
      };
    }
    function te(t) {
      const { properties: e, searchFieldList: n, joinQuery: y, table: u } = t;
      let g = {}, v = 1;
      return Object.keys(e).map((a) => {
        let r = e[a];
        if (r.view == "table") {
          let h = r.properties, x = v * 100;
          Object.keys(h).map((L) => {
            let k = h[L];
            k.order = x + Number(k.order);
            let C = a + "@" + L;
            g[C] = k;
          }), v++;
        } else
          r.order = Number(r.order), g[a] = r;
      }), g;
    }
    function ae() {
      let t = `${l}${i.id}`;
      return new Promise((e) => {
        ye.get({ url: t }, { isTransformResponse: !1 }).then((n) => {
          n.success ? e(n.result) : (e(!1), Y.warning(n.message));
        }).catch(() => {
          Y.warning("获取查询条件失败!"), e(!1);
        });
      });
    }
    const [oe, { resetFields: re, setFieldsValue: A, updateSchema: ne, getFieldsValue: se }] = de({
      schemas: w,
      showActionButtonGroup: !1,
      baseColProps: V,
      autoSubmitOnEnter: !0,
      labelWidth: j,
      wrapperCol: null,
      submitFunc() {
        M();
      }
      /* labelCol: ONL_QUERY_LABEL_COL,
      wrapperCol: ONL_QUERY_WRAPPER_COL*/
    });
    function M() {
      let t = se(), e = Object.assign({}, R(o.param), ie(t));
      s("search", e, !0);
    }
    function le() {
      return D(this, null, function* () {
        yield re();
        const { config: t, param: e } = R(o);
        let n = Object.assign({}, t, e);
        Object.keys(n).length > 0 && (yield A(n)), s("search", n, !1);
      });
    }
    function ie(t) {
      return Object.keys(t).map((e) => {
        t[e] && t[e] instanceof Array && (t[e] = t[e].join(","));
      }), t;
    }
    return Q(
      () => _.value,
      (t) => {
        let e = S.value;
        if (e && e.length > 0) {
          let n = [];
          for (let y of e)
            n.push({
              field: y,
              show: t
            });
          ne(n);
        }
      },
      { immediate: !1 }
    ), {
      onlineQueryFormRef: c,
      registerForm: oe,
      initDefaultValues: m,
      toggleButtonShow: F,
      toggleSearchStatus: _,
      doSearch: M,
      resetSearch: le,
      queryParams: f,
      formSchemas: w
    };
  }
}, P = (i) => (ge("data-v-3e6e229f"), i = i(), _e(), i), De = {
  key: 0,
  class: "jeecg-basic-table-form-container online-query-form p-0"
}, Ie = /* @__PURE__ */ P(() => /* @__PURE__ */ U("span", { class: "group-query-string" }, "~", -1)), Ve = /* @__PURE__ */ P(() => /* @__PURE__ */ U("span", { class: "group-query-string" }, "~", -1)), Ye = /* @__PURE__ */ P(() => /* @__PURE__ */ U("span", { class: "group-query-string" }, "~", -1)), je = {
  style: { float: "left", overflow: "hidden", "margin-left": "10px" },
  class: "table-page-search-submitButtons"
};
function Ce(i, s, l, c, w, V) {
  const F = I("a-date-picker"), _ = I("a-form-item-rest"), f = I("a-input-number"), S = I("a-button"), Y = I("a-icon"), N = I("a-col"), j = I("BasicForm");
  return c.formSchemas && c.formSchemas.length > 0 ? (W(), G("div", De, [
    p(j, {
      ref: "onlineQueryFormRef",
      onRegister: c.registerForm
    }, {
      groupDate: b(({ model: o, field: m }) => [
        p(F, {
          showTime: !1,
          valueFormat: "YYYY-MM-DD",
          placeholder: "开始日期",
          value: o[m + "_begin"],
          "onUpdate:value": (d) => o[m + "_begin"] = d,
          style: { width: "calc(50% - 15px)", "min-width": "100px" }
        }, null, 8, ["value", "onUpdate:value"]),
        Ie,
        p(_, null, {
          default: b(() => [
            p(F, {
              showTime: !1,
              valueFormat: "YYYY-MM-DD",
              placeholder: "结束日期",
              value: o[m + "_end"],
              "onUpdate:value": (d) => o[m + "_end"] = d,
              style: { width: "calc(50% - 15px)", "min-width": "100px" }
            }, null, 8, ["value", "onUpdate:value"])
          ]),
          _: 2
        }, 1024)
      ]),
      groupDatetime: b(({ model: o, field: m }) => [
        p(F, {
          showTime: !0,
          valueFormat: "YYYY-MM-DD HH:mm:ss",
          placeholder: "开始时间",
          value: o[m + "_begin"],
          "onUpdate:value": (d) => o[m + "_begin"] = d,
          style: { "min-width": "100px", width: "calc(50% - 15px)" }
        }, null, 8, ["value", "onUpdate:value"]),
        Ve,
        p(_, null, {
          default: b(() => [
            p(F, {
              showTime: !0,
              valueFormat: "YYYY-MM-DD HH:mm:ss",
              placeholder: "结束时间",
              value: o[m + "_end"],
              "onUpdate:value": (d) => o[m + "_end"] = d,
              style: { "min-width": "100px", width: "calc(50% - 15px)" }
            }, null, 8, ["value", "onUpdate:value"])
          ]),
          _: 2
        }, 1024)
      ]),
      groupNumber: b(({ model: o, field: m }) => [
        p(f, {
          placeholder: "开始值",
          value: o[m + "_begin"],
          "onUpdate:value": (d) => o[m + "_begin"] = d,
          style: { width: "calc(50% - 15px)" }
        }, null, 8, ["value", "onUpdate:value"]),
        Ye,
        p(_, null, {
          default: b(() => [
            p(f, {
              placeholder: "结束值",
              value: o[m + "_end"],
              "onUpdate:value": (d) => o[m + "_end"] = d,
              style: { width: "calc(50% - 15px)" }
            }, null, 8, ["value", "onUpdate:value"])
          ]),
          _: 2
        }, 1024)
      ]),
      formFooter: b(() => [
        p(N, {
          md: 6,
          sm: 8
        }, {
          default: b(() => [
            U("span", je, [
              p(S, {
                preIcon: "ant-design:search",
                type: "primary",
                onClick: c.doSearch
              }, {
                default: b(() => [
                  T("查询")
                ]),
                _: 1
              }, 8, ["onClick"]),
              p(S, {
                preIcon: "ant-design:reload",
                type: "primary",
                onClick: c.resetSearch,
                style: { "margin-left": "8px" }
              }, {
                default: b(() => [
                  T("重置")
                ]),
                _: 1
              }, 8, ["onClick"]),
              c.toggleButtonShow ? (W(), G("a", {
                key: 0,
                onClick: s[0] || (s[0] = (o) => c.toggleSearchStatus = !c.toggleSearchStatus),
                style: { "margin-left": "8px" }
              }, [
                T(he(c.toggleSearchStatus ? "收起" : "展开") + " ", 1),
                p(Y, {
                  type: c.toggleSearchStatus ? "up" : "down"
                }, null, 8, ["type"])
              ])) : K("", !0)
            ])
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["onRegister"])
  ])) : K("", !0);
}
const ht = /* @__PURE__ */ Oe(Le, [["render", Ce], ["__scopeId", "data-v-3e6e229f"]]);
export {
  ht as default
};
