// @jeecg/online@3.6.2-beta/OnlineAutoList-0583a910.mjs
var T = (o, f, l) => new Promise((I, m) => {
  var b = (a) => {
    try {
      r(l.next(a));
    } catch (c) {
      m(c);
    }
  }, S = (a) => {
    try {
      r(l.throw(a));
    } catch (c) {
      m(c);
    }
  }, r = (a) => a.done ? I(a.value) : Promise.resolve(a.value).then(b, S);
  r((l = l.apply(o, f)).next());
});
import { ref as vo, watch as Do, resolveComponent as s, openBlock as t, createElementBlock as C, normalizeClass as Q, createVNode as p, createBlock as u, withCtx as i, createTextVNode as g, createCommentVNode as d, Fragment as _, renderList as Lo, toDisplayString as B, withDirectives as Oo, vShow as Fo, createElementVNode as F, mergeProps as Ao, toHandlers as Po, resolveDynamicComponent as Qo, normalizeProps as _o, guardReactiveProps as Ho } from "vue";
import { BasicTable as Jo, TableAction as $o } from "/@/components/Table";
import { useMessage as qo } from "/@/hooks/web/useMessage";
import No from "./OnlineAutoModal-510991eb.mjs";
import zo from "./OnlineCustomModal-518ad33c.mjs";
import Vo from "./OnlineDetailModal-9675390d.mjs";
import jo from "/@/components/Form/src/jeecg/components/JImportModal.vue";
import { u as Ko, a as Uo } from "./useListButton-cdbab702.mjs";
import { O as Wo, u as Go, a as Xo, g as Yo } from "./useExtendComponent-325fe3f1.mjs";
import Zo from "./OnlineQueryForm-1bfc8320.mjs";
import xo from "./SuperQuery-5d1ca957.mjs";
import { u as oe } from "./useOnlinePopEvent-687070b7.mjs";
import { N as ee } from "./constant-754f1a9d.mjs";
import { _ as ne } from "./index-0e3ab3af.mjs";
import "/@/components/Modal";
import "./OnlineForm-1fa2d893.mjs";
import "/@/components/Form/index";
import "/@/utils/http/axios";
import "lodash-es";
import "/@/utils";
import "/@/components/Loading";
import "/@/components/jeecg/JVxeTable/types";
import "/@/utils/auth";
import "@ant-design/icons-vue";
import "/@/hooks/core/useContext";
import "/@/utils/mitt";
import "./useCustomHook-acb00837.mjs";
import "/@/utils/cache";
import "/@/utils/common/compUtils";
import "/@/store/modules/user";
import "/@/components/jeecg/comment/CommentPanel.vue";
import "./OnlineFormDetail-0e7294ca.mjs";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "vue-router";
import "/@/router";
import "/@/hooks/core/onMountedOrActivated";
import "/@/store/modules/multipleTab";
import "./cgformState-d9f8ec42.mjs";
import "pinia";
import "/@/store";
import "ant-design-vue";
import "/@/hooks/system/useMethods";
import "/@/components/Form/src/componentMap";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/hooks/system/useListPage";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "@vueuse/core";
import "./SuperQueryValComponent.vue_vue_type_script_lang-e9196096.mjs";
import "/@/utils/is";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const te = {
  name: "OnlineAutoList",
  components: {
    BasicTable: Jo,
    TableAction: $o,
    OnlineAutoModal: No,
    JImportModal: jo,
    OnlineQueryForm: Zo,
    OnlineSuperQuery: xo,
    OnlineCustomModal: zo,
    OnlineDetailModal: Vo,
    OnlinePopModal: Wo
  },
  setup() {
    const { createMessage: o } = qo(), {
      ID: f,
      onlineTableContext: l,
      onlineQueryFormOuter: I,
      loading: m,
      reload: b,
      dataSource: S,
      pagination: r,
      handleSpecialConfig: a,
      getColumnList: c,
      handleChangeInTable: M,
      loadData: R,
      superQueryButtonRef: h,
      superQueryStatus: E,
      handleSuperQuery: v,
      onlineExtConfigJson: w,
      handleFormConfig: D,
      registerCustomModal: e,
      tableReloading: y
    } = Ko();
    if (!f.value)
      throw o.warning("地址错误, 配置ID不存在!"), new Error("地址错误, 配置ID不存在!");
    let { initCgEnhanceJs: L } = Go(l);
    const {
      buttonSwitch: H,
      cgLinkButtonList: J,
      cgTopButtonList: $,
      importUrl: q,
      registerModal: N,
      handleAdd: z,
      handleEdit: V,
      handleBatchDelete: j,
      registerImportModal: K,
      onImportExcel: U,
      onExportExcel: A,
      cgButtonJsHandler: W,
      cgButtonActionHandler: G,
      cgButtonLinkHandler: X,
      handleSubmitFlow: Y,
      getDropDownActions: Z,
      getActions: x,
      initButtonList: oo,
      initButtonSwitch: eo,
      registerDetailModal: no,
      registerBpmModal: to
    } = Uo(l, w), O = vo(!1);
    function io() {
      return T(this, null, function* () {
        try {
          O.value = !0, yield A();
        } finally {
          setTimeout(() => O.value = !1, 1500);
        }
      });
    }
    const {
      columns: ro,
      actionColumn: lo,
      selectedKeys: ao,
      rowSelection: so,
      enableScrollBar: uo,
      tableScroll: po,
      downloadRowFile: mo,
      getImgView: co,
      getPcaText: go,
      getFormatDate: fo,
      handleColumnResult: ho,
      hrefComponent: yo,
      viewOnlineCellImage: Co,
      hrefMainTableId: bo,
      registerOnlineHrefModal: So,
      registerPopModal: wo,
      openPopModal: ko,
      onlinePopModalRef: Io,
      popTableId: P
    } = Xo(l, w);
    Do(
      f,
      () => {
        To();
      },
      { immediate: !0 }
    );
    function To() {
      return T(this, null, function* () {
        m.value = !0;
        let n = yield c(ee);
        Bo(n), yield R(), m.value = !1, l.execButtonEnhance("setup");
      });
    }
    function Bo(n) {
      let k = L(n.enhanceJs);
      l.EnhanceJS = k, oo(n.cgButtonList), eo(n.hideColumns), ho(n), a(n);
    }
    function Mo(n) {
      l.queryParam = n, b({ mode: "search" });
    }
    function Ro(n) {
      return T(this, null, function* () {
        yield Yo(h), h.value.init(n);
      });
    }
    function Eo(n) {
      P.value = n.id;
      let k = {
        title: n.describe
      };
      n.record && n.record.id && (k.record = n.record, k.isUpdate = !0), ko(!0, k);
    }
    return oe(Eo), {
      ID: f,
      // 查询区域
      onlineQueryFormOuter: I,
      queryWithCondition: Mo,
      onQueryFormLoaded: Ro,
      reload: b,
      //高级查询
      superQueryButtonRef: h,
      superQueryStatus: E,
      handleSuperQuery: v,
      // table区域
      loading: m,
      columns: ro,
      dataSource: S,
      pagination: r,
      actionColumn: lo,
      rowSelection: so,
      selectedKeys: ao,
      tableScroll: po,
      enableScrollBar: uo,
      handleChangeInTable: M,
      //按钮
      buttonSwitch: H,
      handleAdd: z,
      handleEdit: V,
      onImportExcel: U,
      onExportExcel: A,
      exportLoading: O,
      onExportExcelOverride: io,
      cgTopButtonList: $,
      cgLinkButtonList: J,
      cgButtonJsHandler: W,
      cgButtonActionHandler: G,
      cgButtonLinkHandler: X,
      handleBatchDelete: j,
      // table-slot
      downloadRowFile: mo,
      getImgView: co,
      getPcaText: go,
      getFormatDate: fo,
      // 操作列
      getActions: x,
      getDropDownActions: Z,
      // 弹窗
      registerModal: N,
      registerCustomModal: e,
      registerImportModal: K,
      registerDetailModal: no,
      importUrl: q,
      handleFormConfig: D,
      onlinePopModalRef: Io,
      //其他
      tableReloading: y,
      handleSubmitFlow: Y,
      hrefComponent: yo,
      viewOnlineCellImage: Co,
      hrefMainTableId: bo,
      onlineExtConfigJson: w,
      registerOnlineHrefModal: So,
      registerPopModal: wo,
      popTableId: P,
      registerBpmModal: to
    };
  }
  // 1引入了loadsh   console.log(that.simpleDateFormat(new Date().getTime(),'yyyy-MM-dd'));
  // 2. value的问题
  // 3. 变量位置改变后需要 重写api
  // 1添加按钮的时候 预留出样式对象 然后js增强中设置样式对象
  // 2直接设置css字符串 然后通过js document 往head里面增加css片段 全局生效
};
const ie = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, re = {
  key: 0,
  style: { "font-size": "12px", "font-style": "italic" }
}, le = ["src", "onClick"], ae = ["innerHTML"], se = ["title"];
function ue(o, f, l, I, m, b) {
  const S = s("online-query-form"), r = s("a-button"), a = s("online-super-query"), c = s("TableAction"), M = s("BasicTable"), R = s("OnlineAutoModal"), h = s("online-detail-modal"), E = s("JImportModal"), v = s("a-modal"), w = s("online-custom-modal"), D = s("online-pop-modal");
  return t(), C("div", {
    class: Q(["p-2", `online-list-${o.ID}`])
  }, [
    p(S, {
      ref: "onlineQueryFormOuter",
      id: o.ID,
      onSearch: o.queryWithCondition,
      onLoaded: o.onQueryFormLoaded
    }, null, 8, ["id", "onSearch", "onLoaded"]),
    o.tableReloading ? d("", !0) : (t(), u(M, {
      key: 0,
      ref: "onlineTable",
      rowKey: "jeecg_row_key",
      canResize: !0,
      bordered: !0,
      showIndexColumn: !1,
      loading: o.loading,
      columns: o.columns,
      dataSource: o.dataSource,
      pagination: o.pagination,
      rowSelection: o.rowSelection,
      actionColumn: o.actionColumn,
      showTableSetting: !0,
      clickToRowSelect: !1,
      scroll: o.tableScroll,
      onTableRedo: o.reload,
      class: Q({ "j-table-force-nowrap": o.enableScrollBar }),
      onChange: o.handleChangeInTable
    }, {
      tableTitle: i(() => [
        o.buttonSwitch.add ? (t(), u(r, {
          key: 0,
          preIcon: "ant-design:plus-outlined",
          type: "primary",
          onClick: o.handleAdd
        }, {
          default: i(() => [
            g("新增")
          ]),
          _: 1
        }, 8, ["onClick"])) : d("", !0),
        o.buttonSwitch.import ? (t(), u(r, {
          key: 1,
          preIcon: "ant-design:import-outlined",
          type: "primary",
          onClick: o.onImportExcel
        }, {
          default: i(() => [
            g("导入")
          ]),
          _: 1
        }, 8, ["onClick"])) : d("", !0),
        o.buttonSwitch.export ? (t(), u(r, {
          key: 2,
          preIcon: "ant-design:export-outlined",
          type: "primary",
          loading: o.exportLoading,
          onClick: o.onExportExcelOverride
        }, {
          default: i(() => [
            g(" 导出")
          ]),
          _: 1
        }, 8, ["loading", "onClick"])) : d("", !0),
        o.cgTopButtonList && o.cgTopButtonList.length > 0 ? (t(!0), C(_, { key: 3 }, Lo(o.cgTopButtonList, (e, y) => (t(), C(_, null, [
          e.optType == "js" ? (t(), u(r, {
            key: "cgbtn" + y,
            onClick: (L) => o.cgButtonJsHandler(e.buttonCode),
            type: "primary",
            preIcon: e.buttonIcon ? "ant-design:" + e.buttonIcon : ""
          }, {
            default: i(() => [
              g(B(e.buttonName), 1)
            ]),
            _: 2
          }, 1032, ["onClick", "preIcon"])) : e.optType == "action" ? (t(), u(r, {
            key: "cgbtn" + y,
            onClick: (L) => o.cgButtonActionHandler(e.buttonCode),
            type: "primary",
            preIcon: e.buttonIcon ? "ant-design:" + e.buttonIcon : ""
          }, {
            default: i(() => [
              g(B(e.buttonName), 1)
            ]),
            _: 2
          }, 1032, ["onClick", "preIcon"])) : d("", !0)
        ], 64))), 256)) : d("", !0),
        o.buttonSwitch.batch_delete ? Oo((t(), u(r, {
          key: 4,
          preIcon: "ant-design:delete-outlined",
          onClick: o.handleBatchDelete
        }, {
          default: i(() => [
            g(" 批量删除")
          ]),
          _: 1
        }, 8, ["onClick"])), [
          [Fo, o.selectedKeys.length > 0]
        ]) : d("", !0),
        o.buttonSwitch.super_query ? (t(), u(a, {
          key: 5,
          ref: "superQueryButtonRef",
          online: "",
          status: o.superQueryStatus,
          onSearch: o.handleSuperQuery
        }, null, 8, ["status", "onSearch"])) : d("", !0)
      ]),
      fileSlot: i(({ text: e }) => [
        e ? (t(), u(r, {
          key: 1,
          ghost: !0,
          type: "primary",
          preIcon: "ant-design:download",
          size: "small",
          onClick: (y) => o.downloadRowFile(e)
        }, {
          default: i(() => [
            g(" 下载 ")
          ]),
          _: 2
        }, 1032, ["onClick"])) : (t(), C("span", ie, "无文件"))
      ]),
      imgSlot: i(({ text: e }) => [
        e ? (t(), C("img", {
          key: 1,
          src: o.getImgView(e),
          alt: "图片不存在",
          class: "online-cell-image",
          onClick: (y) => o.viewOnlineCellImage(e)
        }, null, 8, le)) : (t(), C("span", re, "无图片"))
      ]),
      htmlSlot: i(({ text: e }) => [
        F("div", { innerHTML: e }, null, 8, ae)
      ]),
      pcaSlot: i(({ text: e }) => [
        F("div", {
          title: o.getPcaText(e)
        }, B(o.getPcaText(e)), 9, se)
      ]),
      dateSlot: i(({ text: e }) => [
        F("span", null, B(o.getFormatDate(e)), 1)
      ]),
      action: i(({ record: e }) => [
        p(c, {
          actions: o.getActions(e),
          dropDownActions: o.getDropDownActions(e)
        }, null, 8, ["actions", "dropDownActions"])
      ]),
      _: 1
    }, 8, ["loading", "columns", "dataSource", "pagination", "rowSelection", "actionColumn", "scroll", "onTableRedo", "class", "onChange"])),
    p(R, {
      id: o.ID,
      onRegister: o.registerModal,
      onSuccess: o.reload,
      onFormConfig: o.handleFormConfig
    }, null, 8, ["id", "onRegister", "onSuccess", "onFormConfig"]),
    p(h, {
      id: o.ID,
      onRegister: o.registerDetailModal
    }, null, 8, ["id", "onRegister"]),
    p(E, {
      onRegister: o.registerImportModal,
      url: o.importUrl(),
      onOk: o.reload,
      online: ""
    }, null, 8, ["onRegister", "url", "onOk"]),
    p(v, Ao(o.hrefComponent.model, Po(o.hrefComponent.on)), {
      default: i(() => [
        (t(), u(Qo(o.hrefComponent.is), _o(Ho(o.hrefComponent.params)), null, 16))
      ]),
      _: 1
    }, 16),
    p(w, {
      onRegister: o.registerCustomModal,
      onSuccess: o.reload
    }, null, 8, ["onRegister", "onSuccess"]),
    p(h, {
      id: o.hrefMainTableId,
      onRegister: o.registerOnlineHrefModal,
      defaultFullscreen: !1
    }, null, 8, ["id", "onRegister"]),
    p(D, {
      ref: "onlinePopModalRef",
      id: o.popTableId,
      onRegister: o.registerPopModal,
      onSuccess: o.reload,
      request: "",
      topTip: ""
    }, null, 8, ["id", "onRegister", "onSuccess"])
  ], 2);
}
const Tn = /* @__PURE__ */ ne(te, [["render", ue]]);
export {
  Tn as default
};
