// @jeecg/online@3.6.2-beta/OnlineTabFormDetail-5a70219d.mjs
var G = Object.defineProperty;
var K = Object.getOwnPropertySymbols;
var X = Object.prototype.hasOwnProperty, Y = Object.prototype.propertyIsEnumerable;
var V = (a, r, n) => r in a ? G(a, r, { enumerable: !0, configurable: !0, writable: !0, value: n }) : a[r] = n, E = (a, r) => {
  for (var n in r || (r = {}))
    X.call(r, n) && V(a, n, r[n]);
  if (K)
    for (var n of K(r))
      Y.call(r, n) && V(a, n, r[n]);
  return a;
};
var v = (a, r, n) => new Promise((o, g) => {
  var S = (m) => {
    try {
      s(n.next(m));
    } catch (f) {
      g(f);
    }
  }, h = (m) => {
    try {
      s(n.throw(m));
    } catch (f) {
      g(f);
    }
  }, s = (m) => m.done ? o(m.value) : Promise.resolve(m.value).then(S, h);
  s((n = n.apply(a, r)).next());
});
import { useMessage as Z } from "/@/hooks/web/useMessage";
import { ref as u, watch as $, reactive as ee, resolveComponent as c, openBlock as p, createElementBlock as _, createVNode as y, createCommentVNode as N, withCtx as B, createBlock as k, Fragment as te, renderList as oe, normalizeStyle as ne, renderSlot as re } from "vue";
import { Loading as ie } from "/@/components/Loading";
import { getToken as ae } from "/@/utils/auth";
import { goJmReportViewPage as le } from "/@/utils";
import { PrinterOutlined as me } from "@ant-design/icons-vue";
import se from "./DetailForm-424638eb.mjs";
import pe from "./OnlineSubFormDetail-f0d2c016.mjs";
import { i as ue } from "./useExtendComponent-325fe3f1.mjs";
import { defHttp as ce } from "/@/utils/http/axios";
import { E as fe, T as de } from "./constant-754f1a9d.mjs";
import { _ as be } from "./index-0e3ab3af.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/utils/common/compUtils";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "/@/components/Form/index";
import "/@/components/Form/src/componentMap";
import "/@/components/Modal";
import "lodash-es";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/components/Table";
import "/@/hooks/system/useListPage";
import "vue-router";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "ant-design-vue";
import "@vueuse/core";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const ge = {
  name: "OnlineTabFormDetail",
  components: {
    DetailForm: se,
    Loading: ie,
    PrinterOutlined: me,
    OnlineSubFormDetail: pe
  },
  props: {
    id: {
      type: String,
      default: ""
    },
    formTemplate: {
      type: Number,
      default: 1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    isTree: {
      type: Boolean,
      default: !1
    },
    pidField: {
      type: String,
      default: ""
    },
    submitTip: {
      type: Boolean,
      default: !0
    },
    showSub: {
      type: Boolean,
      default: !0
    },
    themeTemplate: {
      type: String,
      default: ""
    },
    tabIndex: {
      type: String,
      default: ""
    }
  },
  emits: ["success", "rendered"],
  setup(a, { emit: r }) {
    const { createMessage: n } = Z(), o = u(""), g = u(!0), S = u(!1), h = u(1), s = u({}), m = u(300), f = u(340), T = u("0");
    $(
      () => a.tabIndex,
      (e, t) => {
        T.value = e, t && J();
      },
      {
        immediate: !0
      }
    );
    const b = ee({
      reportPrintShow: 0,
      reportPrintUrl: "",
      joinQuery: 0,
      modelFullscreen: 0,
      modalMinWidth: ""
    }), { detailFormSchemas: x, hasSubTable: w, subTabInfo: l, refMap: P, showStatus: F, subDataSource: C, createFormSchemas: R, formSpan: A } = ue(a);
    function I(e) {
      let t = { reportPrintShow: 0, reportPrintUrl: "", joinQuery: 0, modelFullscreen: 1, modalMinWidth: "" };
      e && (t = JSON.parse(e)), Object.keys(t).map((i) => {
        b[i] = t[i];
      });
    }
    function M(e) {
      return v(this, null, function* () {
        h.value = e.head.tableType, o.value = e.head.tableName, g.value = e.head.tableType == 1, I(e.head.extConfigJson), R(e.schema.properties), r("rendered", b);
      });
    }
    function H(e, t) {
      return v(this, null, function* () {
        yield U(t), D(!0);
      });
    }
    function L(e) {
      let t = `/online/cgform/api/detail/${a.id}/${e}`;
      return new Promise((i, d) => {
        ce.get({ url: t }, { isTransformResponse: !1 }).then((O) => {
          O.success ? i(O.result) : (d(), n.warning(O.message));
        }).catch(() => {
          d();
        });
      });
    }
    function D(e) {
      Object.keys(F).map((t) => {
        F[t] = e;
      });
    }
    function J() {
      D(!1), setTimeout(() => {
        D(!0);
      }, 300);
    }
    function U(e) {
      return v(this, null, function* () {
        let t = yield L(e.id);
        s.value = E({}, t), j(t);
      });
    }
    function j(e) {
      e || (e = {});
      let t = Object.keys(C.value);
      if (t && t.length > 0) {
        let i = {};
        for (let d of t)
          i[d] = e[d] || [];
        C.value = i;
      }
    }
    function z(e) {
      return "online_" + e + ":";
    }
    function Q() {
      let e = b.reportPrintUrl, t = s.value;
      if (t) {
        let i = t.id, d = ae();
        le(e, i, d);
      }
    }
    function W(e) {
      let t = s.value;
      return q(t, e);
    }
    function q(e, t) {
      if (e) {
        let i = e[t];
        return !i && i !== 0 && (i = e[t.toLowerCase()], !i && i !== 0 && (i = e[t.toUpperCase()])), i;
      }
      return "";
    }
    return {
      detailFormSchemas: x,
      formData: s,
      formSpan: A,
      //主表
      tableName: o,
      loading: S,
      //子表
      hasSubTable: w,
      subTabInfo: l,
      subFormHeight: m,
      subTableHeight: f,
      refMap: P,
      onTabChange: J,
      //一对多子表
      subDataSource: C,
      getSubTableAuthPre: z,
      //父组件调用
      show: H,
      createRootProperties: M,
      // 扩展配置
      onOpenReportPrint: Q,
      onlineExtConfigJson: b,
      getSubTableForeignKeyValue: W,
      showStatus: F,
      ERP: fe,
      TAB: de,
      subActiveKey: T
    };
  }
};
const he = ["id"], _e = {
  key: 0,
  style: { "text-align": "right", position: "absolute", top: "15px", right: "20px", "z-index": "999" }
}, ye = { key: 1 };
function Se(a, r, n, o, g, S) {
  const h = c("PrinterOutlined"), s = c("detail-form"), m = c("a-tab-pane"), f = c("online-sub-form-detail"), T = c("JVxeTable"), b = c("a-spin"), x = c("a-tabs"), w = c("Loading");
  return p(), _("div", {
    id: o.tableName + "_form"
  }, [
    o.formData.id && o.onlineExtConfigJson.reportPrintShow ? (p(), _("div", _e, [
      y(h, {
        title: "打印",
        onClick: o.onOpenReportPrint,
        style: { "font-size": "16px" }
      }, null, 8, ["onClick"])
    ])) : N("", !0),
    y(x, {
      class: "tabTheme",
      onChange: o.onTabChange,
      activeKey: o.subActiveKey,
      "onUpdate:activeKey": r[0] || (r[0] = (l) => o.subActiveKey = l)
    }, {
      default: B(() => [
        (p(), k(m, {
          tab: "主表",
          key: "-1"
        }, {
          default: B(() => [
            y(s, {
              schemas: o.detailFormSchemas,
              data: o.formData,
              span: o.formSpan
            }, null, 8, ["schemas", "data", "span"])
          ]),
          _: 1
        })),
        o.hasSubTable && n.showSub ? (p(!0), _(te, { key: 0 }, oe(o.subTabInfo, (l, P) => (p(), k(m, {
          tab: l.describe,
          key: P + "",
          forceRender: !0
        }, {
          default: B(() => [
            l.relationType == 1 ? (p(), _("div", {
              key: 0,
              style: ne({ "overflow-y": "auto", "overflow-x": "hidden", "max-height": o.subFormHeight + "px" })
            }, [
              y(f, {
                table: l.key,
                "form-template": n.formTemplate,
                "main-id": o.getSubTableForeignKeyValue(l.foreignKey),
                properties: l.properties
              }, null, 8, ["table", "form-template", "main-id", "properties"])
            ], 4)) : (p(), _("div", ye, [
              o.showStatus[l.key] ? (p(), k(T, {
                key: 0,
                ref_for: !0,
                ref: o.refMap[l.key],
                toolbar: "",
                "keep-source": "",
                "row-number": "",
                "row-selection": "",
                height: o.subTableHeight,
                disabled: !0,
                columns: l.columns,
                dataSource: o.subDataSource[l.key],
                authPre: o.getSubTableAuthPre(l.key)
              }, null, 8, ["height", "columns", "dataSource", "authPre"])) : (p(), k(b, {
                key: 1,
                spinning: !0
              }))
            ]))
          ]),
          _: 2
        }, 1032, ["tab"]))), 128)) : N("", !0)
      ]),
      _: 1
    }, 8, ["onChange", "activeKey"]),
    y(w, {
      loading: o.loading,
      absolute: !1
    }, null, 8, ["loading"]),
    re(a.$slots, "bottom", {}, void 0, !0)
  ], 8, he);
}
const mt = /* @__PURE__ */ be(ge, [["render", Se], ["__scopeId", "data-v-e0b07b2d"]]);
export {
  mt as default
};
