// @jeecg/online@3.6.2-beta/OnlineFormDetail-0e7294ca.mjs
var q = Object.defineProperty;
var O = Object.getOwnPropertySymbols;
var G = Object.prototype.hasOwnProperty, X = Object.prototype.propertyIsEnumerable;
var J = (a, i, n) => i in a ? q(a, i, { enumerable: !0, configurable: !0, writable: !0, value: n }) : a[i] = n, V = (a, i) => {
  for (var n in i || (i = {}))
    G.call(i, n) && J(a, n, i[n]);
  if (O)
    for (var n of O(i))
      X.call(i, n) && J(a, n, i[n]);
  return a;
};
var T = (a, i, n) => new Promise((o, g) => {
  var S = (m) => {
    try {
      s(n.next(m));
    } catch (u) {
      g(u);
    }
  }, b = (m) => {
    try {
      s(n.throw(m));
    } catch (u) {
      g(u);
    }
  }, s = (m) => m.done ? o(m.value) : Promise.resolve(m.value).then(S, b);
  s((n = n.apply(a, i)).next());
});
import { useMessage as Y } from "/@/hooks/web/useMessage";
import { ref as d, reactive as Z, resolveComponent as c, openBlock as p, createElementBlock as y, createVNode as k, createCommentVNode as R, createBlock as w, withCtx as B, Fragment as $, renderList as ee, normalizeStyle as te, renderSlot as oe } from "vue";
import { Loading as ne } from "/@/components/Loading";
import { getToken as re } from "/@/utils/auth";
import { goJmReportViewPage as ie } from "/@/utils";
import { PrinterOutlined as ae } from "@ant-design/icons-vue";
import le from "./DetailForm-424638eb.mjs";
import me from "./OnlineSubFormDetail-f0d2c016.mjs";
import { i as se } from "./useExtendComponent-325fe3f1.mjs";
import { defHttp as pe } from "/@/utils/http/axios";
import { E as ce } from "./constant-754f1a9d.mjs";
import { _ as ue } from "./index-0e3ab3af.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Form/src/utils/Area";
import "/@/utils/common/compUtils";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "/@/components/Form/index";
import "/@/components/Form/src/componentMap";
import "/@/components/Modal";
import "lodash-es";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/components/Table";
import "/@/hooks/system/useListPage";
import "vue-router";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "ant-design-vue";
import "@vueuse/core";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const fe = {
  name: "OnlineFormDetail",
  components: {
    DetailForm: le,
    Loading: ne,
    PrinterOutlined: ae,
    OnlineSubFormDetail: me
  },
  props: {
    id: {
      type: String,
      default: ""
    },
    formTemplate: {
      type: Number,
      default: 1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    isTree: {
      type: Boolean,
      default: !1
    },
    pidField: {
      type: String,
      default: ""
    },
    submitTip: {
      type: Boolean,
      default: !0
    },
    showSub: {
      type: Boolean,
      default: !0
    },
    themeTemplate: {
      type: String,
      default: ""
    }
  },
  emits: ["success", "rendered"],
  setup(a, { emit: i }) {
    const { createMessage: n } = Y(), o = d(""), g = d(!0), S = d(!1), b = d(1), s = d({}), m = d(300), u = d(340), h = Z({
      reportPrintShow: 0,
      reportPrintUrl: "",
      joinQuery: 0,
      modelFullscreen: 0,
      modalMinWidth: ""
    }), { detailFormSchemas: P, hasSubTable: x, subTabInfo: C, refMap: l, showStatus: _, subDataSource: F, createFormSchemas: E, formSpan: N } = se(a);
    function M(e) {
      let t = { reportPrintShow: 0, reportPrintUrl: "", joinQuery: 0, modelFullscreen: 1, modalMinWidth: "" };
      e && (t = JSON.parse(e)), Object.keys(t).map((r) => {
        h[r] = t[r];
      });
    }
    function H(e) {
      return T(this, null, function* () {
        b.value = e.head.tableType, o.value = e.head.tableName, g.value = e.head.tableType == 1, M(e.head.extConfigJson), E(e.schema.properties), i("rendered", h);
      });
    }
    function L(e, t) {
      return T(this, null, function* () {
        yield z(t), v(!0);
      });
    }
    function j(e) {
      let t = `/online/cgform/api/detail/${a.id}/${e}`;
      return new Promise((r, f) => {
        pe.get({ url: t }, { isTransformResponse: !1 }).then((D) => {
          D.success ? r(D.result) : (f(), n.warning(D.message));
        }).catch(() => {
          f();
        });
      });
    }
    function v(e) {
      Object.keys(_).map((t) => {
        _[t] = e;
      });
    }
    function U() {
      v(!1), setTimeout(() => {
        v(!0);
      }, 300);
    }
    function z(e) {
      return T(this, null, function* () {
        let t = yield j(e.id);
        s.value = V({}, t), I(t);
      });
    }
    function I(e) {
      e || (e = {});
      let t = Object.keys(F.value);
      if (t && t.length > 0) {
        let r = {};
        for (let f of t)
          r[f] = e[f] || [];
        F.value = r;
      }
    }
    function K(e) {
      return "online_" + e + ":";
    }
    function A() {
      let e = h.reportPrintUrl, t = s.value;
      if (t) {
        let r = t.id, f = re();
        ie(e, r, f);
      }
    }
    function Q(e) {
      let t = s.value;
      return W(t, e);
    }
    function W(e, t) {
      if (e) {
        let r = e[t];
        return !r && r !== 0 && (r = e[t.toLowerCase()], !r && r !== 0 && (r = e[t.toUpperCase()])), r;
      }
      return "";
    }
    return {
      detailFormSchemas: P,
      formData: s,
      formSpan: N,
      //主表
      tableName: o,
      loading: S,
      //子表
      hasSubTable: x,
      subTabInfo: C,
      subFormHeight: m,
      subTableHeight: u,
      refMap: l,
      onTabChange: U,
      //一对多子表
      subDataSource: F,
      getSubTableAuthPre: K,
      //父组件调用
      show: L,
      createRootProperties: H,
      // 扩展配置
      onOpenReportPrint: A,
      onlineExtConfigJson: h,
      getSubTableForeignKeyValue: Q,
      showStatus: _,
      ERP: ce
    };
  }
}, de = ["id"], he = {
  key: 0,
  style: { "text-align": "right", position: "absolute", top: "15px", right: "20px", "z-index": "999" }
}, ge = { key: 1 };
function be(a, i, n, o, g, S) {
  const b = c("PrinterOutlined"), s = c("detail-form"), m = c("online-sub-form-detail"), u = c("JVxeTable"), h = c("a-spin"), P = c("a-tab-pane"), x = c("a-tabs"), C = c("Loading");
  return p(), y("div", {
    id: o.tableName + "_form"
  }, [
    o.formData.id && o.onlineExtConfigJson.reportPrintShow ? (p(), y("div", he, [
      k(b, {
        title: "打印",
        onClick: o.onOpenReportPrint,
        style: { "font-size": "16px" }
      }, null, 8, ["onClick"])
    ])) : R("", !0),
    k(s, {
      schemas: o.detailFormSchemas,
      data: o.formData,
      span: o.formSpan
    }, null, 8, ["schemas", "data", "span"]),
    n.themeTemplate !== o.ERP && o.hasSubTable && n.showSub ? (p(), w(x, {
      key: 1,
      onChange: o.onTabChange
    }, {
      default: B(() => [
        (p(!0), y($, null, ee(o.subTabInfo, (l, _) => (p(), w(P, {
          tab: l.describe,
          key: _ + "",
          forceRender: !0
        }, {
          default: B(() => [
            l.relationType == 1 ? (p(), y("div", {
              key: 0,
              style: te({ "overflow-y": "auto", "overflow-x": "hidden", "max-height": o.subFormHeight + "px" })
            }, [
              k(m, {
                table: l.key,
                "form-template": n.formTemplate,
                "main-id": o.getSubTableForeignKeyValue(l.foreignKey),
                properties: l.properties
              }, null, 8, ["table", "form-template", "main-id", "properties"])
            ], 4)) : (p(), y("div", ge, [
              o.showStatus[l.key] ? (p(), w(u, {
                key: 0,
                ref_for: !0,
                ref: o.refMap[l.key],
                toolbar: "",
                "keep-source": "",
                "row-number": "",
                "row-selection": "",
                height: o.subTableHeight,
                disabled: !0,
                columns: l.columns,
                dataSource: o.subDataSource[l.key],
                authPre: o.getSubTableAuthPre(l.key)
              }, null, 8, ["height", "columns", "dataSource", "authPre"])) : (p(), w(h, {
                key: 1,
                spinning: !0
              }))
            ]))
          ]),
          _: 2
        }, 1032, ["tab"]))), 128))
      ]),
      _: 1
    }, 8, ["onChange"])) : R("", !0),
    k(C, {
      loading: o.loading,
      absolute: !1
    }, null, 8, ["loading"]),
    oe(a.$slots, "bottom")
  ], 8, de);
}
const it = /* @__PURE__ */ ue(fe, [["render", be]]);
export {
  it as default
};
