// @jeecg/online@3.6.2-beta/OnlineSubFormDetail-f0d2c016.mjs
var g = (t, i, r) => new Promise((o, m) => {
  var c = (e) => {
    try {
      n(r.next(e));
    } catch (s) {
      m(s);
    }
  }, a = (e) => {
    try {
      n(r.throw(e));
    } catch (s) {
      m(s);
    }
  }, n = (e) => e.done ? o(e.value) : Promise.resolve(e.value).then(c, a);
  n((r = r.apply(t, i)).next());
});
import { useMessage as F } from "/@/hooks/web/useMessage";
import { ref as u, watch as d, resolveComponent as S, openBlock as v, createBlock as D } from "vue";
import { BasicForm as w } from "/@/components/Form/index";
import { defHttp as y } from "/@/utils/http/axios";
import { i as $, g as x } from "./useExtendComponent-325fe3f1.mjs";
import { Loading as B } from "/@/components/Loading";
import I from "./DetailForm-424638eb.mjs";
import { _ as O } from "./index-0e3ab3af.mjs";
import "/@/components/Form/src/componentMap";
import "/@/utils/propTypes";
import "@ant-design/icons-vue";
import "/@/components/Modal";
import "lodash-es";
import "./constant-754f1a9d.mjs";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/common/compUtils";
import "/@/components/Table";
import "/@/hooks/system/useListPage";
import "vue-router";
import "/@/components/Form/src/utils/Area";
import "/@/components/Preview/index";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "/@/utils/auth";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "ant-design-vue";
import "@vueuse/core";
import "/@/utils/dict";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/components/Markdown";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const R = "/online/cgform/api/subform", T = {
  name: "OnlineSubFormDetail",
  components: {
    BasicForm: w,
    Loading: B,
    DetailForm: I
  },
  props: {
    properties: {
      type: Object,
      required: !0
    },
    mainId: {
      type: String,
      default: ""
    },
    table: {
      type: String,
      default: ""
    },
    formTemplate: {
      type: Number,
      default: 1
    }
  },
  emits: ["formChange"],
  setup(t) {
    const i = u(!1), { createMessage: r } = F(), o = u(""), m = u({}), { detailFormSchemas: c, createFormSchemas: a, formSpan: n } = $(t);
    d(
      () => t.table,
      () => {
        o.value = t.table;
      },
      { immediate: !0 }
    ), d(
      () => t.properties,
      () => {
        i.value = !1, a(t.properties), i.value = !0;
      },
      { deep: !0, immediate: !0 }
    ), d(
      () => t.mainId,
      () => {
        setTimeout(() => {
          e();
        }, 100);
      },
      { immediate: !0 }
    );
    function e() {
      return g(this, null, function* () {
        yield x(i), m.value = {};
        const { table: p, mainId: l } = t;
        if (!p || !l)
          return;
        let f = (yield s(p, l)) || {};
        m.value = f;
      });
    }
    function s(p, l) {
      let f = `${R}/${p}/${l}`;
      return new Promise((_, h) => {
        y.get({ url: f }, { isTransformResponse: !1 }).then((b) => {
          b.success ? _(b.result) : h();
        });
      }).catch(() => {
        r.warning("子表获取数据失败:" + p);
      });
    }
    return {
      detailFormSchemas: c,
      subFormData: m,
      formSpan: n
    };
  }
};
function k(t, i, r, o, m, c) {
  const a = S("detail-form");
  return v(), D(a, {
    schemas: o.detailFormSchemas,
    data: o.subFormData,
    span: o.formSpan
  }, null, 8, ["schemas", "data", "span"]);
}
const Ft = /* @__PURE__ */ O(T, [["render", k]]);
export {
  Ft as default
};
