// @jeecg/online@3.6.2-beta/OnlineTabDetailModal-7356d3a8.mjs
var F = (e, o, a) => new Promise((s, i) => {
  var u = (t) => {
    try {
      n(a.next(t));
    } catch (r) {
      i(r);
    }
  }, p = (t) => {
    try {
      n(a.throw(t));
    } catch (r) {
      i(r);
    }
  }, n = (t) => t.done ? s(t.value) : Promise.resolve(t.value).then(u, p);
  n((a = a.apply(e, o)).next());
});
import { defineComponent as z, ref as C, watch as G, resolveComponent as d, openBlock as I, createBlock as M, mergeProps as J, withCtx as m, createElementVNode as N, toDisplayString as R, createVNode as c, createElementBlock as K, Fragment as Q, renderList as X, createTextVNode as S } from "vue";
import { BasicModal as Y } from "/@/components/Modal";
import Z from "./OnlineTabFormDetail-5a70219d.mjs";
import { b as ee } from "./useExtendComponent-325fe3f1.mjs";
import te from "/@/components/jeecg/comment/CommentPanel.vue";
import { T as oe } from "./constant-754f1a9d.mjs";
import { _ as ae } from "./index-0e3ab3af.mjs";
import "/@/hooks/web/useMessage";
import "/@/components/Loading";
import "/@/utils/auth";
import "/@/utils";
import "@ant-design/icons-vue";
import "./DetailForm-424638eb.mjs";
import "/@/utils/propTypes";
import "/@/utils/dict";
import "/@/utils/dict/JDictSelectUtil";
import "/@/utils/dict/index";
import "/@/api/common/api";
import "/@/utils/http/axios";
import "/@/components/Form/src/utils/Area";
import "/@/utils/common/compUtils";
import "/@/components/Preview/index";
import "/@/components/Markdown";
import "./OnlineSubFormDetail-f0d2c016.mjs";
import "/@/components/Form/index";
import "/@/components/Form/src/componentMap";
import "lodash-es";
import "/@/components/Form/src/jeecg/components/JUpload";
import "/@/views/system/user/user.api";
import "/@/store/modules/user";
import "/@/utils/desform/customExpression";
import "/@/store/modules/permission";
import "/@/components/Table";
import "/@/hooks/system/useListPage";
import "vue-router";
import "./LinkTableListPiece-8499a318.mjs";
import "./OnlineSelectCascade-4265c95d.mjs";
import "./JModalTip-33c7dc44.mjs";
import "/@/hooks/web/useAppInject";
import "ant-design-vue";
import "@vueuse/core";
import "/@/components/jeecg/OnLine/JPopupOnlReport.vue";
const ne = z({
  name: "OnlineTabDetailModal",
  props: {
    id: {
      type: String,
      required: !1,
      default: ""
    }
  },
  components: {
    BasicModal: Y,
    OnlineFormDetail: Z,
    CommentPanel: te
  },
  emits: ["success", "register", "formConfig"],
  setup(e, { emit: o }) {
    const a = C(), s = C([]), i = C("-1");
    function u() {
      a.value && a.value.reload();
    }
    const {
      title: p,
      modalWidth: n,
      registerModal: t,
      cgButtonList: r,
      handleCgButtonClick: f,
      disableSubmit: h,
      handleSubmit: l,
      submitLoading: _,
      handleCancel: k,
      handleFormConfig: D,
      onlineFormCompRef: y,
      formTemplate: V,
      isTreeForm: O,
      pidFieldName: $,
      renderSuccess: E,
      formRendered: P,
      showSub: A,
      tableName: L,
      formDataId: j,
      enableComment: W,
      themeTemplate: q
    } = ee(!1, { emit: o }, u);
    G(() => e.id, H, { immediate: !0 });
    function H() {
      return F(this, null, function* () {
        P.value = !1, e.id && (yield D(e.id, {}, (g) => {
          const w = [], T = [], { head: B, schema: U } = g, { properties: x } = U;
          w.push({ tableName: B.tableName, tableTxt: B.tableTxt }), Object.entries(x).forEach(([v, b]) => {
            b.view == "tab" && T.push({ tableName: v, tableTxt: b.describe });
          }), T.sort((v, b) => v.order - b.order), s.value = [...w, ...T];
        }));
      });
    }
    return {
      title: p,
      onlineFormCompRef: y,
      renderSuccess: E,
      registerModal: t,
      handleSubmit: l,
      handleCancel: k,
      modalWidth: n,
      formTemplate: V,
      disableSubmit: h,
      cgButtonList: r,
      handleCgButtonClick: f,
      isTreeForm: O,
      pidFieldName: $,
      submitLoading: _,
      showSub: A,
      tableName: L,
      formDataId: j,
      enableComment: W,
      commentPanelRef: a,
      themeTemplate: q,
      tabNav: s,
      tabValue: (g) => String(g - 1),
      tabIndex: i,
      TAB: oe,
      restTabIndex: () => {
        setTimeout(() => {
          i.value = "-1";
        }, 500);
      }
    };
  }
});
const re = { class: "titleArea" }, me = { class: "title" }, ie = { class: "right" };
function le(e, o, a, s, i, u) {
  const p = d("a-radio-button"), n = d("a-radio-group"), t = d("a-button"), r = d("online-form-detail"), f = d("comment-panel"), h = d("BasicModal");
  return I(), M(h, J({
    title: e.title,
    onCancel: o[2] || (o[2] = () => {
      e.restTabIndex();
    }),
    width: e.modalWidth,
    maxHeight: 600,
    enableComment: e.enableComment,
    defaultFullscreen: !1
  }, e.$attrs, {
    onRegister: e.registerModal,
    wrapClassName: "jeecg-online-detail-modal"
  }), {
    title: m(() => [
      N("div", re, [
        N("div", me, R(e.title), 1),
        N("div", ie, [
          c(n, {
            value: e.tabIndex,
            "onUpdate:value": o[0] || (o[0] = (l) => e.tabIndex = l)
          }, {
            default: m(() => [
              (I(!0), K(Q, null, X(e.tabNav, (l, _) => (I(), M(p, {
                value: e.tabValue(_),
                key: l.tableName
              }, {
                default: m(() => [
                  S(R(l.tableTxt), 1)
                ]),
                _: 2
              }, 1032, ["value"]))), 128))
            ]),
            _: 1
          }, 8, ["value"])
        ])
      ])
    ]),
    footer: m(() => [
      c(t, {
        key: "back",
        onClick: o[1] || (o[1] = () => {
          e.handleCancel(), e.restTabIndex();
        })
      }, {
        default: m(() => [
          S("关闭")
        ]),
        _: 1
      })
    ]),
    comment: m(() => [
      c(f, {
        ref: "commentPanelRef",
        tableName: e.tableName,
        dataId: e.formDataId
      }, null, 8, ["tableName", "dataId"])
    ]),
    default: m(() => [
      c(r, {
        ref: "onlineFormCompRef",
        id: e.id,
        "form-template": e.formTemplate,
        "show-sub": e.showSub,
        themeTemplate: e.themeTemplate,
        tabIndex: e.tabIndex,
        onRendered: e.renderSuccess
      }, null, 8, ["id", "form-template", "show-sub", "themeTemplate", "tabIndex", "onRendered"])
    ]),
    _: 1
  }, 16, ["title", "width", "enableComment", "onRegister"]);
}
const tt = /* @__PURE__ */ ae(ne, [["render", le], ["__scopeId", "data-v-341d8c57"]]);
export {
  tt as default
};
